---
title: "Pocillopora acuta Intraspecific Diversity Effects: Larvae & Adults"
author: "Ariana Huffmyer (R Markdown author), Nina Bean, Casey Harris, Crawford Drury"
date: "2021"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 6
    toc_float: yes
  pdf_document:
    keep_tex: yes
editor_options: 
  chunk_output_type: console
---

## Setup  

Set up workspace, set options, and load required packages.  
```{r, echo=FALSE, show=FALSE}
library("knitr")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```
 
```{r}
library(multcomp)
library(ggplot2)
library(lattice)
library(MASS)
library(car)
library(FSA)
library(partitionBEFsp)
library(factoextra)
library(ggfortify)
library(cowplot)
library(readxl)
library(ggdendro)
library(ape)
library(SNPRelate)
library(tidyverse)
library(magicfor)
library(janitor)
library(readxl)
```
 
# **Genotyping**  

We used next generation sequencing to obtain 150bp paired-end reads of DNA from all parent colonies in larvae, juvenile, and adult experiments. Scripts of sequence analysis are available in the R Project as a text file. After obtaining an identity-by-distance (IBS) matris with the ANGSD package in Python, we produced a tree using the hclust function with "complete" clustering.  

Produce an MDS plot and tree with ANGSD data. Blue line indicates cut-off for clonal groups based on distance between biological replicates.   
```{r, results=TRUE}
#read IBS matrix
#setwd("Data/")
name <- "Data/angsd.IBS.ibsMat"
m <- as.matrix(read.table(name))

#replace labels with sample names
samples<-read.table("Data/sample_list.txt");samples<-as.vector(samples$V1)
colnames(m)<-samples
rownames(m)<-samples

#mds plot
mds <- cmdscale(as.dist(m))
plot(mds,lwd=2,ylab="Dist",xlab="Dist",main="multidimensional scaling",col=rep(1:32))

plot(hclust(dist(m), "complete"))
abline(h=0.01485, col="blue") #genetic distance b/t SM29 and SM41
```

Generate final figure with colors representing clonal groups.   
```{r, results=TRUE, message=FALSE, warning=FALSE}
rawdata<-read_tsv("Data/angsd.IBS.ibsMat",col_names=FALSE)%>%select(-X37)

hcdata<-dendro_data(hclust(dist(rawdata),"complete"),type="rectangle")
hcdata$segments<-hcdata$segments%>%mutate(yend=case_when(yend==0~(y-0.003),TRUE ~ (yend)))
hcdata$labels<-left_join(hcdata$labels,hcdata$segments,by="x")%>%select(-y.x)%>%dplyr::rename(y=yend)%>%mutate(y=y-.01)
hcdata$segments<-left_join(hcdata$segments,hcdata$labels,by="x")%>%select(-y.y,-xend.y,-y.y.y)%>%dplyr::rename(y=y.x,xend=xend.x)%>%
     mutate(color=case_when(label=="8"~"a",
                            label=="31"~"a",
                            label=="29"~"a",
                            label=="7"~"a",
                            label=="10"~"a",
                            label=="17"~"a",
                            label=="32"~"a",
                            label=="11"~"b",
                            label=="13"~"b",
                            label=="18"~"b",
                            label=="5"~"c",
                            label=="6"~"c",
                            label=="1"~"c",
                            label=="4"~"c",
                            label=="28"~"d",
                            label=="12"~"d",
                            label=="16"~"d",
                            TRUE ~ ("e")))

quartz(width=1.75,height=1.5)
Figure1<-ggplot() + 
     geom_hline(yintercept=0.0148,color="gray",linetype="dashed")+
     geom_segment(data=segment(hcdata), aes(x=x, y=y, xend=xend, yend=yend,colour=color),size=.5) +
     #scale_y_continuous(breaks=seq(-.002,.02,.002),limits=c(-.02,.02))+
     theme_classic()+
     theme(axis.ticks.x=element_blank(),
           axis.text.x=element_blank(),
           axis.line.x=element_blank(),
           axis.title.x=element_blank(),
           legend.position="none")+
     ylab("Genetic Dist")+
     scale_color_manual(values=c("red","blue","orange","darkgreen","black"));Figure1
```

```{r, results=FALSE}
#ggsave(filename="Figures/Figure1.pdf", plot=Figure1, dpi=300, width=1.75, height=1.5, units="in")
```
These clonal groups are accounted for in below analysis in which number of genotypes is at times lower than the number of colonies due to clonal groupings.  

# **Larval Experiment**  

In this experiment with larvae, we exposed larvae of *P. acuta* either to single (3 larvae from same parent colony) or multiple (1 larvae from each of 3 different parent colonies) to ambient or high temperature. Larvae were put in 96-well plates and treated with temperature using a heat block. Larvae were not exposed to sunlight or artificial LED light. Genotype indicates genotype determined by above genotyping analysis, this value does not reflect the parent ID.   

## 1. Load and manipulate data sets  

First, load survivorship data frame. Survival was recorded for each larvae at 2 and 4 days of temperature treatment exposure. Data was collected at the level of the individual larva. Here, we also reassign colony identity to clonal identity to account for number of distinct individuals.    

```{r, results=FALSE}
#load data
surv<-read.csv("Data/larval_surv.csv", sep=",", na.strings="NA")
surv$Plate<-as.factor(surv$Plate)
surv$Well<-as.factor(surv$Well)
surv$Column<-as.factor(surv$Column)
surv$Row<-as.factor(surv$Row)
surv$site<-as.factor(surv$site)
surv$colony<-as.factor(surv$colony)
surv$diversity_level<-as.factor(surv$diversity_level)
surv$Genotype<-surv$colony

#assign genotype identification according to parent ID
levels(surv$Genotype)[levels(surv$Genotype)=="831"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="821"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="801"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="841"] <- "ClusterA"

levels(surv$Genotype)[levels(surv$Genotype)=="741"] <- "ClusterB"
levels(surv$Genotype)[levels(surv$Genotype)=="743"] <- "ClusterB"
levels(surv$Genotype)[levels(surv$Genotype)=="802"] <- "ClusterB"

levels(surv$Genotype)[levels(surv$Genotype)=="815"] <- "ClusterD"
levels(surv$Genotype)[levels(surv$Genotype)=="742"] <- "ClusterD"
levels(surv$Genotype)[levels(surv$Genotype)=="750"] <- "ClusterD"

#format column names and add treatment columns 
surv$Treatment <- ifelse(surv$diversity_level %in% c("1"), "Single",
                     ifelse(surv$diversity_level %in% c("3"), "Diverse", NA)) 
surv$Treatment<-as.factor(surv$Treatment)
```

Then, assign Success and Failure integer columns. Success (1) = alive and Failure (0) = dead. Subset data to analyze survivorship at the end of 4-day treatment period.  

```{r, results=FALSE}
#assign outcome as two columns: success and failure
surv$success<-ifelse(surv$outcome==1,1,0)
surv$failure<-ifelse(surv$outcome==0,1,0)

#provide subset for each day
midsurv <- surv[which(surv$duration == "2"),]
finalsurv <- surv[which(surv$duration == "4"),]
```

Load the settlement data frame and assign Success and Failure columns as above for survivorship. Settlement was also recorded at the level of the individual larva and tracked at 2 and 4 days of treatment exposure.   

```{r, results=FALSE}
settle<-read.csv("Data/larval_settle.csv", sep=",", na.strings="NA")
settle$Date<-as.factor(settle$Date)
settle$Plate<-as.factor(settle$Plate)
settle$Well<-as.factor(settle$Well)
settle$Column<-as.factor(settle$Column)
settle$Row<-as.factor(settle$Row)
settle$site<-as.factor(settle$site)
settle$colony<-as.factor(settle$colony)
settle$diversity_level<-as.factor(settle$diversity_level)
settle$Genotype<-settle$colony

settle$success<-ifelse(settle$outcome==0,1,0)
settle$failure<-ifelse(settle$outcome==1,1,0)

levels(settle$Genotype)[levels(settle$Genotype)=="831"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="821"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="801"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="841"] <- "ClusterA"

levels(settle$Genotype)[levels(settle$Genotype)=="741"] <- "ClusterB"
levels(settle$Genotype)[levels(settle$Genotype)=="743"] <- "ClusterB"
levels(settle$Genotype)[levels(settle$Genotype)=="802"] <- "ClusterB"

levels(settle$Genotype)[levels(settle$Genotype)=="815"] <- "ClusterD"
levels(settle$Genotype)[levels(settle$Genotype)=="742"] <- "ClusterD"
levels(settle$Genotype)[levels(settle$Genotype)=="750"] <- "ClusterD"

#format column names and add treatment columns 
settle$Treatment <- ifelse(settle$diversity_level %in% c("1"), "Single",
                     ifelse(settle$diversity_level %in% c("3"), "Diverse", NA)) 

settle$Treatment<-as.factor(settle$Treatment)

#provide subset for each day
midsettle <- settle[which(settle$duration == "2"),]
finalsettle <- settle[which(settle$duration == "4"),]
```
 
## 2. Calculate metric means    

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
finalsurv<-finalsurv %>%
  group_by(Plate, Treatment)%>%
  mutate(N_Genotypes = length(unique(Genotype)))

finalsettle<-finalsettle %>%
  group_by(Plate)%>%
  mutate(N_Genotypes = length(unique(Genotype)))

#survivorship
selection_surv <- plyr::ddply(finalsurv, c("Plate", "Treatment", "Genotype", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(outcome, na.rm=TRUE))

#settlement
selection_settle <- plyr::ddply(finalsettle, c("Plate", "Treatment", "Genotype", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(outcome, na.rm=TRUE))


selection_surv$Metric<-c("Survival")
selection_surv$Stage<-c("Larvae")

selection_settle$Metric<-c("Settlement")
selection_settle$Stage<-c("Larvae")

#output data for use in calculations  

#spread by diversity treatment
selection_surv<-selection_surv %>% 
  spread(Treatment, mean)

selection_settle<-selection_settle %>% 
  spread(Treatment, mean)

#remove Plate column
selection_surv<-selection_surv[-c(1)]
selection_settle<-selection_settle[-c(1)]

write.csv(selection_surv, "Output/larval_survival_means.csv")
write.csv(selection_settle, "Output/larval_settlement_means.csv")
```


# **Recruit Experiment**  

In this experiment with recruits, we examined the survival and growth response of recruit corals fused with additional recruits of either the same, or multiple, parental genotypes (e*P. acuta*). Data are published in Huffmyer et al. (2021) in Coral Reefs and is analyzed here to calculate diversity metrics. Genotype indicates genotype determined by above genotyping analysis, this value does not reflect the parent ID.   

## 1. Load and manipulate data sets  

First, load recruit data frame. Survival was recorded for each recruit in self or non-self fusions throughout a temperature treatment exposure. Data was collected at the level of the individual recruit.     

```{r, results=FALSE}
#load data and extract fused colonies
data<-read.csv("Data/Juvenile_data.csv", sep=",", na.strings="NA")
  
recruits<-data%>%
  filter(Fusion.Type=="Fused")%>%
  select(Day, Slide, Spat, Colony, Richness.Type, Treatment, Fusion.Partners, Survivorship, Polyps)

#assign genotype identification according to parent ID
levels(recruits$Colony)[levels(recruits$Colony)=="831-47"] <- "ClusterA"
levels(recruits$Colony)[levels(recruits$Colony)=="821-47"] <- "ClusterA"
levels(recruits$Colony)[levels(recruits$Colony)=="801-47"] <- "ClusterA"
levels(recruits$Colony)[levels(recruits$Colony)=="841-47"] <- "ClusterA"

levels(recruits$Colony)[levels(recruits$Colony)=="741-F"] <- "ClusterB"
levels(recruits$Colony)[levels(recruits$Colony)=="743-13"] <- "ClusterB"
levels(recruits$Colony)[levels(recruits$Colony)=="802-42"] <- "ClusterB"

levels(recruits$Colony)[levels(recruits$Colony)=="815-25"] <- "ClusterD"
levels(recruits$Colony)[levels(recruits$Colony)=="742-F"] <- "ClusterD"
levels(recruits$Colony)[levels(recruits$Colony)=="750-F"] <- "ClusterD"

recruits$Fusion.Partners<-as.factor(recruits$Fusion.Partners)
recruits$Day<-as.factor(recruits$Day)
recruits$Polyps<-as.numeric(recruits$Polyps)

#format column names and add treatment columns 
recruits$Diversity <- ifelse(recruits$Richness.Type %in% c("Single"), "Single",
                     ifelse(recruits$Richness.Type %in% c("Multiple"), "Diverse", NA)) 

recruits<-recruits%>%
  select(Day, Slide, Spat, Colony, Treatment, Fusion.Partners, Survivorship, Polyps, Diversity)

```

## 2. Calculate metric means    

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
recruit_final<-recruits %>%
  group_by(Day, Slide, Treatment, Diversity)%>%
  mutate(N_Genotypes = length(unique(Colony)))

#survivorship and growth means
recruit_surv <- plyr::ddply(recruit_final, c("Day", "Treatment", "Diversity", "N_Genotypes"), summarise,
                           surv = mean(Survivorship, na.rm=TRUE))

recruit_growth <- plyr::ddply(recruit_final, c("Day", "Treatment", "Diversity", "N_Genotypes"), summarise,
                           polyps = mean(Polyps, na.rm=TRUE))


recruit_surv$Metric<-c("Survival")
recruit_surv$Stage<-c("Recruit")

recruit_growth$Metric<-c("Growth")
recruit_growth$Stage<-c("Recruit")

#spread by diversity treatment
recruit_surv<-recruit_surv %>% 
  spread(Diversity, surv)

recruit_growth<-recruit_growth %>% 
  spread(Diversity, polyps)

write.csv(recruit_surv, "Output/recruit_survival_means.csv")
write.csv(recruit_growth, "Output/recruit_growth_means.csv")
```


# **Adult Experiment**  

Adults in this experiment were cut as fragments and exposed to either single or multiple colony groups. Single groups (n=40, n=400 fragments) included ten fragments of the same colony while multiple genotype groups (n=20 groups, n=200 fragments) included one fragment from each of ten parent colonies. In these calculations, genotype number was adjusted based on results from genetic analysis.   

Adults were exposed to ambient and high temperature treatments in outdoor flow through tank systems under one layer of shade cloth (light peaking at ~250-300 PAR daily). Corals were contained in plastic containers for each group so water was not exchanged. See manuscript for more details of methodology.  

Diversity metrics for survivorship, growth, and photosynthetic efficiency are calculated as described above.  

## 1. Survivorship    
 
Load survivorship data from the stress test period. Survival over the course of the 25 day stress test was recorded at the level of the individual fragment. In calculations we are analyzing survivorship by the end of this period.      
 
```{r, results=FALSE}
#load data
stress<-read.csv("Data/Adults_Survivorship.csv", header=TRUE, sep=",", na.strings="NA")
stress$Genotype<-as.factor(stress$Colony)

#change genotypes to reflect clonal clusters 
levels(stress$Genotype)[levels(stress$Genotype)=="3"] <- "ClusterA"
levels(stress$Genotype)[levels(stress$Genotype)=="32"] <- "ClusterA"
levels(stress$Genotype)[levels(stress$Genotype)=="5"] <- "ClusterA"

levels(stress$Genotype)[levels(stress$Genotype)=="2"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="31"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="12"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="24"] <- "ClusterC"

stress$success<-ifelse(stress$Censor==0,1,0)
stress$failure<-ifelse(stress$Censor==1,1,0)

#assign outcome as two columns: success and failure at the end point
endstress <- stress[which(stress$Day == "107"),]
endstress$success<-ifelse(endstress$Censor==0,1,0)
endstress$failure<-ifelse(endstress$Censor==1,1,0)
adultstress<-cbind(endstress$success, endstress$failure)
```
 
Calculate mean survivorship.   
```{r, results=TRUE}

#add column for unique counts to insert the number of genotypes
endstress<-endstress %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#survivorship
selection_stress <- plyr::ddply(endstress, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(success, na.rm=TRUE))

#spread by diversity treatment
selection_stress<-selection_stress %>% 
  spread(Treatment, mean)

selection_stress$Metric<-c("Survival")
selection_stress$Stage<-c("Adult")

#output data for analysis
write.csv(selection_stress, "Output/adult_survival_means.csv")
```

Plot survivorship for each genotype in each treatment.  
```{r}
surv_summary<-endstress%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot<-ggplot(data=surv_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot

ggsave("Figures/Adult_Surv.pdf", plot=surv_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

Plot by colony.  
```{r}
surv_summary_colony<-endstress%>%
  group_by(Colony, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot_colony<-ggplot(data=surv_summary_colony, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Colony)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot_colony

ggsave("Figures/Adult_Surv_Colony.pdf", plot=surv_plot_colony, height=4, width=16, units = c("in"), dpi=300) #output figure
```

Run Cox proportional hazards model to test for statistical relationships in adult survival (1/0's).  

```{r}
library(lme4)
library(lmerTest)
library(survival)

cox1<-coxph(Surv(Censor)~Temperature * Treatment * Genotype, data=endstress) #check whether succsess or failure should be used here!
summary(cox1)
Anova(cox1, type=2)
```

Generate a dataframe of change in survival between single and diverse groups in each temperature treatment.  
```{r}
colony_summary<-endstress%>%
    select(Treatment, Colony, Temperature, success)%>%
    group_by(Treatment, Colony, Temperature)%>%
    summarise(mean=mean(success))%>%
    spread(Treatment, mean)%>%
    mutate(percent_change=(Diverse-Single)/Diverse)%>%
    write_csv("Output/Colony_Surival_Change.csv")

```


Plot survial as a time series for each genotype.  

```{r}
surv_summary_time<-stress%>%
  group_by(Day, Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot_time<-ggplot(data=surv_summary_time, aes(x = Day, y = mean, color = Temperature, shape=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3)) +
   #geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3) +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=interaction(Temperature, Treatment), linetype=Treatment)) +
  scale_color_manual(values = c("#F8766D", "#00BFC4"), limits=c("High", "Ambient"))+
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    xlab("Time") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot_time

ggsave("Figures/Adult_Surv_Timeseries.pdf", plot=surv_plot_time, height=4, width=12, units = c("in"), dpi=300) #output figure
```

## 2. SOD production  

Load SOD data (E Majerova) and plot correlation with survival. Subset for high temperature responses and plot correlation with SOD for Diverse survival, Single survival, and Percent change survival.  

```{r}
sod_correlation<-colony_summary%>%
  filter(Temperature=="High")

sod_correlation$Colony<-as.factor(sod_correlation$Colony)

sod<-read.csv("Data/SOD_activity.csv")

sod_summary<-sod%>%
  group_by(Colony)%>%
  summarise(SOD=mean(SOD))

sod_summary$Colony<-as.factor(sod_summary$Colony)

correlation<-left_join(sod_correlation, sod_summary)
```

Test for differences in SOD by colony.  

```{r}
summary(aov(SOD~Colony, data=sod))
```

Plot correlation between survival and SOD production.  

```{r}
Cor1<-ggplot(correlation, aes(x=SOD, y=percent_change)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
    theme(legend.position="none")+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
  ylab("% Change Diverse vs Single")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor1

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

```{r}
Cor2<-ggplot(correlation, aes(x=SOD, y=Diverse)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
    theme(legend.position="none")+
    ylab("Survival in Diverse")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor2

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

```{r}
Cor3<-ggplot(correlation, aes(x=SOD, y=Single)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
  theme(legend.position="right")+
  ylab("Survival in Single")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor3

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

Run spearman correlation
```{r}
cor.test(correlation$SOD, correlation$percent_change, method=c("spearman"))
cor.test(correlation$SOD, correlation$Diverse, method=c("spearman"))
cor.test(correlation$SOD, correlation$Single, method=c("spearman"))
```

Generate three panel plot.  

```{r}
sod_corr<-plot_grid(Cor1, Cor2, Cor3, labels = c("", "", ""), ncol=3, nrow=1, rel_heights= c(1,1,1), rel_widths = c(1,1,1.2), label_size = 20, label_y=1, align="h")

ggsave(filename="Figures/SOD_correlations.png", plot=sod_corr, dpi=300, width=20, height=6, units="in")
```

## 3. Calcification  

Calcification (growth) was measured using the buoyant weight technique. We are analyzing growth as total percent growth (grams per gram per day) as reached by the end of the stress test period. 

Load in buoyant data from Excel worksheet. Loading in final calculations, manipulations conducted in Excel worksheet.     

```{r, restults=FALSE}
bwt<-read.csv("Data/Adults_Growth.csv", header=T, na.strings="NA")
bwt$Genotype<-as.factor(bwt$Colony)

#change genotypes to reflect clonal clusters 
levels(bwt$Genotype)[levels(bwt$Genotype)=="3"] <- "ClusterA"
levels(bwt$Genotype)[levels(bwt$Genotype)=="32"] <- "ClusterA"
levels(bwt$Genotype)[levels(bwt$Genotype)=="5"] <- "ClusterA"

levels(bwt$Genotype)[levels(bwt$Genotype)=="2"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="31"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="12"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="24"] <- "ClusterC"

bwt_stress<-bwt[which(bwt$Phase == "Stress"),]
```

Calculate mean values.  

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
bwt_stress<-bwt_stress %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#calculate means
selection_bwt <- plyr::ddply(bwt_stress, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(TotalPercentDay, na.rm=TRUE))

#spread by diversity treatment
selection_bwt<-selection_bwt %>% 
  spread(Treatment, mean)

#output data for analysis
selection_bwt$Metric<-c("Growth")
selection_bwt$Stage<-c("Adult")

write.csv(selection_bwt, "Output/adult_bwt_means.csv")
```


Plot bwt for each genotype in each treatment.  
```{r}
bwt_summary<-bwt_stress%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(TotalPercentDay, na.rm=TRUE), sd=sd(TotalPercentDay, na.rm=TRUE), N=length(TotalPercentDay[!is.na(TotalPercentDay)]), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

bwt_plot<-ggplot(data=bwt_summary, aes(x = Temperature, y = mean, color = Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3)) +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3) +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment)) +
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Growth"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); bwt_plot

ggsave("Figures/Adult_bwt.pdf", plot=bwt_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```


## 4. PAM    

Photosynthetic efficiency (measured by PAM method). This value is collected as maximum quantum yield (Fv/Fm) for each fragment during the stress test. Here, we use calcualtions of PAM at the end of the stress test. 

Load in PAM datasets and match the record number to sample.    

```{r, results=FALSE, warning=FALSE, message=FALSE}
#load sample id and master data
pam<-read.csv("Data/PAM.csv", sep=",", na.strings="NA")
pam$Genotype<-as.factor(pam$Colony)
pam$Date<-as.factor(pam$Date)
pam$Sample<-paste(pam$Date, pam$ID)

#load pam data
t1<-read.csv("Data/T1_61019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t1$Date <- c("20190610")
t1<-plyr::rename(t1, c("V4"="ID", "V10"="FvFm"))
t1$Sample<-paste(t1$Date, t1$ID)

t2<-read.csv("Data/T2_61719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t2$Date <- c("20190617")
t2<-plyr::rename(t2, c("V4"="ID", "V10"="FvFm"))
t2$Sample<-paste(t2$Date, t2$ID)

t3<-read.csv("Data/T3_62019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t3$Date <- c("20190620")
t3<-plyr::rename(t3, c("V4"="ID", "V10"="FvFm"))
t3$Sample<-paste(t3$Date, t3$ID)

t4<-read.csv("Data/T4_62419.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t4$Date <- c("20190624")
t4<-plyr::rename(t4, c("V4"="ID", "V10"="FvFm"))
t4$Sample<-paste(t4$Date, t4$ID)

t5<-read.csv("Data/T5_62719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t5$Date <- c("20190627")
t5<-plyr::rename(t5, c("V4"="ID", "V10"="FvFm"))
t5$Sample<-paste(t5$Date, t5$ID)

t6<-read.csv("Data/T6_70119.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t6$Date <- c("20190701")
t6<-plyr::rename(t6, c("V4"="ID", "V10"="FvFm"))
t6$Sample<-paste(t6$Date, t6$ID)

#combine all pam data files 
data<-rbind(t1,t2,t3,t4,t5, t6)

#match FvFm values by date and sample number
pam$FvFm<-data$FvFm[match(pam$Sample, data$Sample)]
pam$FvFm<-as.numeric(pam$FvFm)

#match colony to genotype cluster
levels(pam$Genotype)[levels(pam$Genotype)=="3"] <- "ClusterA"
levels(pam$Genotype)[levels(pam$Genotype)=="32"] <- "ClusterA"
levels(pam$Genotype)[levels(pam$Genotype)=="5"] <- "ClusterA"

levels(pam$Genotype)[levels(pam$Genotype)=="2"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="31"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="12"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="24"] <- "ClusterC"
```

Calculate a percent change for each fragment over the course of the stress test.      

```{r, results=FALSE, warning=FALSE}
initial<-subset(pam, Date=="20190610", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm, Genotype))
final<-subset(pam, Date=="20190701", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm, Genotype))

totalPAM1<-rbind(initial, final)

totalPAM1<-spread(totalPAM1, StressDay, FvFm)

totalPAM1$Change<-(totalPAM1$`20`-totalPAM1$`1`)/totalPAM1$`1`

#remove colony 23, which only had one frag observation
totalPAM<-totalPAM1[!totalPAM1$Frag == "1_23_14", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_49", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_57", ]

```

Generate mean values for calculations.     

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
totalPAM<-totalPAM %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#calculate means
selection_pam <- plyr::ddply(totalPAM, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(Change, na.rm=TRUE))

#spread by diversity treatment
selection_pam<-selection_pam %>% 
  spread(Treatment, mean)

#output data for analysis
selection_pam$Metric<-c("Photosynthetic Efficiency")
selection_pam$Stage<-c("Adult")

write.csv(selection_pam, "Output/adult_pam_means.csv")
```

Plot survivorship for each genotype in each treatment.  
```{r}
pam_summary<-totalPAM%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(Change, na.rm=TRUE), sd=sd(Change, na.rm=TRUE), N=length(Change[!is.na(Change)]), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

pam_plot<-ggplot(data=pam_summary, aes(x = Temperature, y = mean, color = Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3)) +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3) +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment)) +
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Rel Fv/Fm"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); pam_plot

ggsave("Figures/Adult_PAM.pdf", plot=pam_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

# **Combine all data sets**  
 
```{r}
selection_values<-dplyr::bind_rows(selection_surv, selection_settle)
selection_values<-dplyr::bind_rows(selection_values, selection_stress)
selection_values<-dplyr::bind_rows(selection_values, selection_bwt)
selection_values<-dplyr::bind_rows(selection_values, selection_pam)

#add "experiment" and "dataset" columns to master dataframe for calculations
metadata<-read.csv("Data/metadata.csv", na.strings="NA")
selection_values_all<-left_join(selection_values, metadata)

#rename diverse column (Yoj) and single column (M) for calculations
selection_values_all<-selection_values_all%>%
  rename(Yoj=Diverse)%>%
  rename(M=Single)

write.csv(selection_values_all, "Output/selection_values_all.csv")
``` 




# **Calculate coefficients and produce figure**  

Calculate coefficients for selection, complementarity, and net biodiversity effects. Scripts written by C. Drury.  

```{r}
raw<-read.csv("Output/selection_values_all.csv")%>%mutate(Yoj=as.numeric(Yoj),M=as.numeric(M),experiment=as.factor(experiment))
#where M=uncorrected survivorship in monoculture (proportion) and Yoj=uncorreted survivorship in polyculture
out<-raw%>%group_by(experiment)%>%filter(M!="NA")%>%add_tally(name="N_Genotypes")%>% #filter missing values and recalculate number of genotypes
     mutate(Mi=M*N_Genotypes)%>%
     mutate(Yo=sum(Yoj))%>%
     mutate(Ryej=1/n())%>%
     mutate(Ryoj=Yoj/Mi)%>%
     mutate(Ryoj=case_when(Ryoj==Inf~0,TRUE~as.numeric(Ryoj)))%>% #divide by zero problem in one AHS
     mutate(Yej=Mi*Ryej)%>%
     mutate(Ye=sum(Yej))%>%
     mutate(deltaY=Yo-Ye)%>%
     mutate(deltaRY=Ryoj-Ryej)%>%
     mutate(C=n()*mean(Mi)*mean(deltaRY))%>%
     mutate(S=cov(deltaRY,Mi)*n())%>%
     mutate(N=S+C)%>%
     slice(1)%>%select(experiment,S,C,N)%>%
     left_join(.,raw%>%select(experiment,Temperature,Stage,Metric)%>%distinct(),by="experiment")
```

Build figure of coefficients.  

```{r}
var_width = 20
plot<-out%>%gather(compartment,coefficient,-Temperature,-experiment,-Metric,-Stage)%>%
     group_by(Temperature,Metric,Stage,compartment)%>%mutate(mean=mean(coefficient),sd=sd(coefficient))%>%
     mutate(Metric=str_wrap(Metric, width = var_width))

plot$compartment<-factor(plot$compartment,levels=c("S","C","N"),labels=c("S","C","Net"))

adult<-ggplot(plot%>%filter(Stage=="Adult"))+geom_hline(yintercept=0,color="lightgray",linetype="dashed")+
     geom_point(aes(compartment,mean,color=Temperature),size=2)+geom_errorbar(aes(compartment,ymin=mean-sd,ymax=mean+sd),width=0)+
     facet_wrap(~Metric,scale="free_x")+
     theme_classic(base_size=8)+
     scale_color_manual(values=c("Blue","Red"))+
     theme(axis.title.x=element_blank(),
           legend.key.size=unit(0.2,"cm"),
           axis.text.y=element_blank(),
           axis.line.y=element_blank(),
           axis.ticks.y=element_blank(),
           axis.title.y=element_blank())+
     scale_y_continuous(limits=c(-0.7,1.1),breaks=seq(-0.7,1.1,0.2))

larvae<-ggplot(plot%>%filter(Stage=="Larvae"))+geom_hline(yintercept=0,color="lightgray",linetype="dashed")+
     geom_point(aes(compartment,mean,color=Temperature),position = position_dodge(width = 0.5))+
     geom_errorbar(aes(compartment,ymin=mean-sd,ymax=mean+sd,color=Temperature),position=position_dodge(width=.5),width=0.1)+
     facet_wrap(~Metric)+
     theme_classic(base_size=8)+
     scale_color_manual(values=c("Blue","Red"))+
     ylab("Coefficient")+
     theme(axis.title.x=element_blank(),
           legend.position="none")+
     scale_y_continuous(limits=c(-0.7,1.1),breaks=seq(-0.7,1.1,0.2))

quartz(h=2,w=5.2)
plot_grid(larvae,adult,rel_widths=c(2,3.2))

diversity_fig<-plot_grid(larvae, adult, labels = c("", ""), ncol=2, nrow=1, rel_heights= c(1.2,1), rel_widths = c(1,1.5), label_size = 20, label_y=1, align="h")

ggsave(filename="Figures/Figure2.pdf", plot=diversity_fig, dpi=300, width=5.2, height=2, units="in")

```

 
