---
title: "Pocillopora acuta Intraspecific Diversity Effects"
author: "Ariana S Huffmyer, Crawford Drury"
date: "2021"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 8
    toc_float: yes
  pdf_document:
    keep_tex: yes
editor_options: 
  chunk_output_type: console
---

## **Setup**  

Set up workspace, set options, and load required packages.   
```{r, echo=FALSE, show=FALSE}
library("knitr")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```
 
```{r}
library(multcomp)
library(ggplot2)
library(lattice)
library(MASS)
library(car)
library(FSA)
library(partitionBEFsp)
library(factoextra)
library(ggfortify)
library(cowplot)
library(readxl)
library(ggdendro)
library(ape)
library(SNPRelate)
library(tidyverse)
library(magicfor)
library(janitor)
library(readxl)
library(lme4)
library(lmerTest)
```
 
# **Genotyping**  

We used next generation sequencing to obtain 150bp paired-end reads of DNA from all parent colonies in larvae, juvenile, and adult experiments. Scripts of sequence analysis are available in the R Project as a text file. After obtaining an identity-by-distance (IBS) matris with the ANGSD package in Python, we produced a tree using the hclust function with "complete" clustering.   

Produce an MDS plot and tree with ANGSD data. Blue line indicates cut-off for clonal groups based on distance between biological replicates.    
```{r, results=TRUE}
#read IBS matrix
#setwd("Data/")
name <- "Data/angsd.IBS.ibsMat"
m <- as.matrix(read.table(name))

#replace labels with sample names
samples<-read.table("Data/sample_list.txt");samples<-as.vector(samples$V1)
colnames(m)<-samples
rownames(m)<-samples

#mds plot
mds <- cmdscale(as.dist(m))
plot(mds,lwd=2,ylab="Dist",xlab="Dist",main="multidimensional scaling",col=rep(1:32))

plot(hclust(dist(m), "complete"))
abline(h=0.01485, col="blue") #genetic distance b/t SM29 and SM41
```

Generate final figure with colors representing clonal groups.   
```{r, results=TRUE, message=FALSE, warning=FALSE}
rawdata<-read_tsv("Data/angsd.IBS.ibsMat",col_names=FALSE)%>%select(-X37)

hcdata<-dendro_data(hclust(dist(rawdata),"complete"),type="rectangle")
hcdata$segments<-hcdata$segments%>%mutate(yend=case_when(yend==0~(y-0.003),TRUE ~ (yend)))
hcdata$labels<-left_join(hcdata$labels,hcdata$segments,by="x")%>%select(-y.x)%>%dplyr::rename(y=yend)%>%mutate(y=y-.01)
hcdata$segments<-left_join(hcdata$segments,hcdata$labels,by="x")%>%select(-y.y,-xend.y,-y.y.y)%>%dplyr::rename(y=y.x,xend=xend.x)%>%
     mutate(color=case_when(label=="8"~"a",
                            label=="31"~"a",
                            label=="29"~"a",
                            label=="7"~"a",
                            label=="10"~"a",
                            label=="17"~"a",
                            label=="32"~"a",
                            label=="11"~"b",
                            label=="13"~"b",
                            label=="18"~"b",
                            label=="5"~"c",
                            label=="6"~"c",
                            label=="1"~"c",
                            label=="4"~"c",
                            label=="28"~"d",
                            label=="12"~"d",
                            label=="16"~"d",
                            TRUE ~ ("e")))

quartz(width=1.75,height=1.5)
Figure1<-ggplot() + 
     geom_hline(yintercept=0.0148,color="gray",linetype="dashed")+
     geom_segment(data=segment(hcdata), aes(x=x, y=y, xend=xend, yend=yend,colour=color),size=.5) +
     #scale_y_continuous(breaks=seq(-.002,.02,.002),limits=c(-.02,.02))+
     theme_classic()+
     theme(axis.ticks.x=element_blank(),
           axis.text.x=element_blank(),
           axis.line.x=element_blank(),
           axis.title.x=element_blank(),
           legend.position="none")+
     ylab("Genetic Dist")+
     scale_color_manual(values=c("red","blue","orange","darkgreen","black"));Figure1
```

```{r, results=FALSE}
#ggsave(filename="Figures/Figure1.pdf", plot=Figure1, dpi=300, width=1.75, height=1.5, units="in")
```
These clonal groups are accounted for in below analysis in which number of genotypes is at times lower than the number of colonies due to clonal groupings.  

# **Larval Experiment**  

In this experiment with larvae, we exposed larvae of *P. acuta* either to single (3 larvae from same parent colony) or multiple (1 larvae from each of 3 different parent colonies) to ambient or high temperature. Larvae were put in 96-well plates and treated with temperature using a heat block. Larvae were not exposed to sunlight or artificial LED light. Genotype indicates genotype determined by above genotyping analysis, this value does not reflect the parent ID.    

## 1. Load and manipulate data sets   

First, load survivorship data frame. Survival was recorded for each larvae at 2 and 4 days of temperature treatment exposure. Data was collected at the level of the individual larva. Here, we also reassign colony identity to clonal identity to account for number of distinct individuals.     
 
```{r, results=FALSE}
#load data
surv<-read.csv("Data/larval_surv.csv", sep=",", na.strings="NA")
surv$Plate<-as.factor(surv$Plate)
surv$Well<-as.factor(surv$Well)
surv$Column<-as.factor(surv$Column)
surv$Row<-as.factor(surv$Row)
surv$site<-as.factor(surv$site)
surv$colony<-as.factor(surv$colony)
surv$diversity_level<-as.factor(surv$diversity_level)
surv$Genotype<-surv$colony

#assign genotype identification according to parent ID
levels(surv$Genotype)[levels(surv$Genotype)=="831"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="821"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="801"] <- "ClusterA"
levels(surv$Genotype)[levels(surv$Genotype)=="841"] <- "ClusterA"

levels(surv$Genotype)[levels(surv$Genotype)=="741"] <- "ClusterB"
levels(surv$Genotype)[levels(surv$Genotype)=="743"] <- "ClusterB"
levels(surv$Genotype)[levels(surv$Genotype)=="802"] <- "ClusterB"

levels(surv$Genotype)[levels(surv$Genotype)=="815"] <- "ClusterD"
levels(surv$Genotype)[levels(surv$Genotype)=="742"] <- "ClusterD"
levels(surv$Genotype)[levels(surv$Genotype)=="750"] <- "ClusterD"

#format column names and add treatment columns 
surv$Treatment <- ifelse(surv$diversity_level %in% c("1"), "Single",
                     ifelse(surv$diversity_level %in% c("3"), "Diverse", NA)) 
surv$Treatment<-as.factor(surv$Treatment)
```

Then, assign Success and Failure integer columns. Success (1) = alive and Failure (0) = dead. Subset data to analyze survivorship at the end of 4-day treatment period.  

```{r, results=FALSE}
#assign outcome as two columns: success and failure
surv$success<-ifelse(surv$outcome==1,1,0)
surv$failure<-ifelse(surv$outcome==0,1,0)

#provide subset for each day
midsurv <- surv[which(surv$duration == "2"),]
finalsurv <- surv[which(surv$duration == "4"),]
```

Load the settlement data frame and assign Success and Failure columns as above for survivorship. Settlement was also recorded at the level of the individual larva and tracked at 2 and 4 days of treatment exposure.   

```{r, results=FALSE}
settle<-read.csv("Data/larval_settle.csv", sep=",", na.strings="NA")
settle$Date<-as.factor(settle$Date)
settle$Plate<-as.factor(settle$Plate)
settle$Well<-as.factor(settle$Well)
settle$Column<-as.factor(settle$Column)
settle$Row<-as.factor(settle$Row)
settle$site<-as.factor(settle$site)
settle$colony<-as.factor(settle$colony)
settle$diversity_level<-as.factor(settle$diversity_level)
settle$Genotype<-settle$colony

settle$success<-ifelse(settle$outcome==0,1,0)
settle$failure<-ifelse(settle$outcome==1,1,0)

levels(settle$Genotype)[levels(settle$Genotype)=="831"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="821"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="801"] <- "ClusterA"
levels(settle$Genotype)[levels(settle$Genotype)=="841"] <- "ClusterA"

levels(settle$Genotype)[levels(settle$Genotype)=="741"] <- "ClusterB"
levels(settle$Genotype)[levels(settle$Genotype)=="743"] <- "ClusterB"
levels(settle$Genotype)[levels(settle$Genotype)=="802"] <- "ClusterB"

levels(settle$Genotype)[levels(settle$Genotype)=="815"] <- "ClusterD"
levels(settle$Genotype)[levels(settle$Genotype)=="742"] <- "ClusterD"
levels(settle$Genotype)[levels(settle$Genotype)=="750"] <- "ClusterD"

#format column names and add treatment columns 
settle$Treatment <- ifelse(settle$diversity_level %in% c("1"), "Single",
                     ifelse(settle$diversity_level %in% c("3"), "Diverse", NA)) 

settle$Treatment<-as.factor(settle$Treatment)

#provide subset for each day
midsettle <- settle[which(settle$duration == "2"),]
finalsettle <- settle[which(settle$duration == "4"),]
```
 
## 2. Calculate metric means     

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
finalsurv<-finalsurv %>%
  group_by(Plate, Treatment)%>%
  mutate(N_Genotypes = length(unique(Genotype)))

finalsettle<-finalsettle %>%
  group_by(Plate)%>%
  mutate(N_Genotypes = length(unique(Genotype)))

#survivorship
selection_surv <- plyr::ddply(finalsurv, c("Plate", "Treatment", "Genotype", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(outcome, na.rm=TRUE))

#settlement
selection_settle <- plyr::ddply(finalsettle, c("Plate", "Treatment", "Genotype", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(outcome, na.rm=TRUE))


selection_surv$Metric<-c("Survival")
selection_surv$Stage<-c("Larvae")

selection_settle$Metric<-c("Settlement")
selection_settle$Stage<-c("Larvae")

#output data for use in calculations  

#spread by diversity treatment
selection_surv<-selection_surv %>% 
  spread(Treatment, mean)

selection_settle<-selection_settle %>% 
  spread(Treatment, mean)

#remove Plate column
selection_surv<-selection_surv[-c(1)]
selection_settle<-selection_settle[-c(1)]

write.csv(selection_surv, "Output/larval_survival_means.csv")
write.csv(selection_settle, "Output/larval_settlement_means.csv")
```

## 3. Plotting  

Plot larval survivorship.  
```{r}
larv_surv_summary<-selection_surv%>%
  gather(key="Treatment", Diverse:Single, value="surv")%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(surv), sd=sd(surv), N=length(surv), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

larv_surv_plot<-ggplot(data=larv_surv_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylim(0, 1)+
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); larv_surv_plot

ggsave("Figures/Larval_Surv.pdf", plot=larv_surv_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

Plot larval settlement.  
```{r}
larv_settle_summary<-selection_settle%>%
  gather(key="Treatment", Diverse:Single, value="settle")%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(settle), sd=sd(settle), N=length(settle), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

larv_settle_plot<-ggplot(data=larv_settle_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylim(0.5, 1)+
    ylab(expression(bold(paste("Settlement Rate"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); larv_settle_plot

ggsave("Figures/Larval_Settle.pdf", plot=larv_settle_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

## 4. Analysis  

### Survival  

Analyze survival with a binomial mixed effect model.    

```{r}
model_larv_surv<-glmer(success~Temperature * Treatment * Genotype + (1|Plate), data=finalsurv, family="binomial") 
summary(model_larv_surv)
Anova(model_larv_surv, type=2) 
```

Analyze model including Genotype as a random effect due to repeated measures.  
```{r}
model_larv_surv2<-glmer(success~Temperature * Treatment * Genotype + (1|Plate) + (1|Genotype), data=finalsurv, family="binomial") 
summary(model_larv_surv2)
Anova(model_larv_surv2, type=2) 
```

Analyze model with genotype only as a random effect.   
```{r}
model_larv_surv3<-glmer(success~Temperature * Treatment + (1|Plate) + (1|Genotype), data=finalsurv, family="binomial") 
summary(model_larv_surv3)
Anova(model_larv_surv3, type=2) 
```


### Settlement  

Conduct linear mixed effect model to analyze the effect of temperature treatment, diversity, and genotype on growth.  

```{r}
model_settle<-glmer(success~Temperature * Treatment * Genotype + (1|Plate), data=finalsettle, family="binomial")
summary(model_settle)
Anova(model_settle, type=2)
```

Add genotype as a random effect to account for repeated measures.  
```{r}
model_settle_2<-glmer(success~Temperature * Treatment * Genotype + (1|Plate) + (1|Genotype), data=finalsettle, family="binomial")
summary(model_settle_2)
Anova(model_settle_2, type=2)
```

Analyze with genotype only as a random effect.  
```{r}
model_settle_3<-glmer(success~Temperature * Treatment + (1|Plate) + (1|Genotype), data=finalsettle, family="binomial")
summary(model_settle_3)
Anova(model_settle_3, type=2)
```  

## 5. Generate full larval figure  

```{r}
larvalfig<-plot_grid(larv_surv_plot, larv_settle_plot, labels=c("A", "B"), label_size=22, ncol=1, nrow=2, rel_heights= c(1,1), rel_widths = c(1,1), align="vh")

ggsave(filename="Figures/Larval_Figure.png", plot=larvalfig, dpi=500, width=14, height=8, units="in")
```


# **Adult Experiment**  

*Notes from 12/3*: We see significant effects of temperature when we analyze with temperature x treatment and having genotype as only a random effect. We can discuss how we want to analyze. For larval survival, temperature and treatment and the interaction were all significant. For settlement, only treatment was significant.  
  
Adults in this experiment were cut as fragments and exposed to either single or multiple colony groups. Single groups (n=40, n=400 fragments) included ten fragments of the same colony while multiple genotype groups (n=20 groups, n=200 fragments) included one fragment from each of ten parent colonies. In these calculations, genotype number was adjusted based on results from genetic analysis.   
Adults were exposed to ambient and high temperature treatments in outdoor flow through tank systems under one layer of shade cloth (light peaking at ~250-300 PAR daily). Corals were contained in plastic containers for each group so water was not exchanged. See manuscript for more details of methodology.  

Diversity metrics for survivorship, growth, and photosynthetic efficiency are calculated as described above.   

## 1. Survivorship    
 
Load survivorship data from the stress test period. Survival over the course of the 25 day stress test was recorded at the level of the individual fragment. In calculations we are analyzing survivorship by the end of this period.       
  
```{r, results=FALSE}
#load data
stress<-read.csv("Data/Adults_Survivorship.csv", header=TRUE, sep=",", na.strings="NA")
stress$Genotype<-as.factor(stress$Colony)

#change genotypes to reflect clonal clusters 
levels(stress$Genotype)[levels(stress$Genotype)=="3"] <- "ClusterA"
levels(stress$Genotype)[levels(stress$Genotype)=="32"] <- "ClusterA"
levels(stress$Genotype)[levels(stress$Genotype)=="5"] <- "ClusterA"

levels(stress$Genotype)[levels(stress$Genotype)=="2"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="31"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="12"] <- "ClusterC"
levels(stress$Genotype)[levels(stress$Genotype)=="24"] <- "ClusterC"

stress$success<-ifelse(stress$Censor==0,1,0)
stress$failure<-ifelse(stress$Censor==1,1,0)

#assign outcome as two columns: success and failure at the end point
endstress <- stress[which(stress$Day == "107"),]
endstress$success<-ifelse(endstress$Censor==0,1,0)
endstress$failure<-ifelse(endstress$Censor==1,1,0)
adultstress<-cbind(endstress$success, endstress$failure)
```
 
Calculate mean survivorship.   
```{r, results=TRUE}

#add column for unique counts to insert the number of genotypes
endstress<-endstress %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#survivorship
selection_stress <- plyr::ddply(endstress, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(success, na.rm=TRUE))

#spread by diversity treatment
selection_stress<-selection_stress %>% 
  spread(Treatment, mean)

selection_stress$Metric<-c("Survival")
selection_stress$Stage<-c("Adult")

#output data for analysis
write.csv(selection_stress, "Output/adult_survival_means.csv")
```

### Plotting  

Plot survivorship for each genotype in each treatment.  
```{r}
surv_summary<-endstress%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot<-ggplot(data=surv_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot

ggsave("Figures/Adult_Surv.pdf", plot=surv_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

Plot by colony.  
```{r}
surv_summary_colony<-endstress%>%
  group_by(Colony, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot_colony<-ggplot(data=surv_summary_colony, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Colony)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot_colony

#ggsave("Figures/Adult_Surv_Colony.pdf", plot=surv_plot_colony, height=4, width=16, units = c("in"), dpi=300) #output figure
```

### Analysis  
 
Analyze survival with a cox proportional hazards model.   

```{r}
library(survival)
model_surv<-coxph(Surv(Censor)~Temperature * Treatment * Genotype, data=endstress)
anova(model_surv, type=2)
``` 
 
Analyze survival using a binaomial model with success/failure vectors which will allow us to incorporate random effects. Plate (repeated measures), day (variable not of interest as a main effect), and genotype (repeated measures), and frag (repeated measures) are used as random effects.   

```{r}
attach(stress)

y<-cbind(success, failure) #generate binomial vector

model_surv_b<-glmer(y~Temperature * Treatment * Genotype + (1|Plate) + (1|Day) + (1|Genotype/Frag) + (1|Table), family=binomial, data=stress) 

detach(stress)
```

View significance of model.  
```{r}
Anova(model_surv_b, type="II")
```

In this model, we have genotype as a significant effect, but no treatment effects. This is a very different result than the Cox Proportional Hazards Model above. Discuss this.  

Attempt another model including table as a random effect.  
 
```{r}
attach(endstress)

y<-cbind(success, failure) #generate binomial vector

model_surv_c<-glmer(y~Temperature * Treatment * Genotype + (1|Plate) + (1|Genotype) + (1|Table), family=binomial, data=endstress) 

detach(endstress)
```

View significance of model.  
```{r}
Anova(model_surv_c, type="II")
```

All models that include random effects only have genotype as significant without any effects of treatment or temperature.  

Model with genotype only as a random effect, not a fixed effect.  

```{r}
attach(endstress)

y<-cbind(success, failure) #generate binomial vector

model_surv_d<-glmer(y~Temperature * Treatment + (1|Plate) + (1|Genotype) + (1|Table), family=binomial, data=endstress) 

detach(endstress)
```

View significance of model.  
```{r}
Anova(model_surv_d, type="II")
```

Temperature is significant.  

Generate a dataframe of change in survival between single and diverse groups in each temperature treatment.  
```{r}
colony_summary<-endstress%>%
    select(Treatment, Colony, Temperature, success)%>%
    group_by(Treatment, Colony, Temperature)%>%
    summarise(mean=mean(success))%>%
    spread(Treatment, mean)%>%
    mutate(percent_change=(Diverse-Single)/Diverse)%>%
    write_csv("Output/Colony_Surival_Change.csv")

geno_summary<-endstress%>%
    select(Treatment, Genotype, Temperature, success)%>%
    group_by(Treatment, Genotype, Temperature)%>%
    summarise(mean=mean(success))%>%
    spread(Treatment, mean)%>%
    mutate(percent_change=(Diverse-Single)/Diverse)%>%
    write_csv("Output/Genotype_Surival_Change.csv")

```


Plot survial as a time series for each genotype.   

```{r}
surv_summary_time<-stress%>%
  group_by(Day, Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(success), sd=sd(success), N=length(success), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

surv_plot_time<-ggplot(data=surv_summary_time, aes(x = Day, y = mean, color = Temperature, shape=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3)) +
   #geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3) +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=interaction(Temperature, Treatment), linetype=Treatment)) +
  scale_color_manual(values = c("#F8766D", "#00BFC4"), limits=c("High", "Ambient"))+
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    xlab("Time") + 
    ylab(expression(bold(paste("Survival"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); surv_plot_time

#ggsave("Figures/Adult_Surv_Timeseries.pdf", plot=surv_plot_time, height=4, width=12, units = c("in"), dpi=300) #output figure
```

## 2. SOD production  

Load superoxide dismutase activity data (E Majerova) and plot correlation with survival. Subset for high temperature responses and plot correlation with SOD for Diverse survival, Single survival, and Percent change survival.   

```{r}
sod_correlation<-colony_summary%>%
  filter(Temperature=="High")

sod_correlation$Colony<-as.factor(sod_correlation$Colony)

sod<-read.csv("Data/SOD_activity.csv")

sod_summary<-sod%>%
  group_by(Colony)%>%
  summarise(SOD=mean(SOD))

sod_summary$Colony<-as.factor(sod_summary$Colony)

correlation<-left_join(sod_correlation, sod_summary)
```

Test for differences in SOD by colony.  

```{r}
kruskal.test(SOD~Colony,data=sod)
```

Significant differences in SOD by colony.  

Plot SOD by colony.  

```{r}
sod_plot<-sod%>%ggplot(aes(x=as.factor(Colony), y=SOD, group=Colony))+
  geom_boxplot()+
  geom_point(color="darkgray")+
  theme_classic()+
  xlab("Colony");sod_plot

ggsave("Figures/Colongy_SOD.png", plot=sod_plot, height=6, width=8, units = c("in"), dpi=300) #output figure
```

Plot correlation between survival and SOD production.  

```{r}
Cor1<-ggplot(correlation, aes(x=SOD, y=percent_change)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
    theme(legend.position="none")+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
  ylab("% Change Diverse vs Single")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor1

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

```{r}
Cor2<-ggplot(correlation, aes(x=SOD, y=Diverse)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
    theme(legend.position="none")+
    ylab("Survival in Diverse")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor2

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

```{r}
Cor3<-ggplot(correlation, aes(x=SOD, y=Single)) + 
  geom_point(aes(color=Colony), size=6) +
  theme_classic()+
  #geom_text(x=800, y=20, label="r=0.50, p<0.001", size=9, color="black") +
  theme(text = element_text(size = 20))+
  theme(legend.position="right")+
  ylab("Survival in Single")+
  theme(axis.text = element_text(size = 20, color="black"))+
  theme(axis.title = element_text(size = 20, face="bold"))+
  theme(legend.title = element_text(size = 20, face="bold"))+
  theme(legend.text = element_text(size = 20, color="black")); Cor3

#ggsave("Figures/Correlation_SOD.png", plot=Cor1, height=10, width=12, units = c("in"), dpi=300) #output figure
```

Run spearman correlation
```{r}
cor.test(correlation$SOD, correlation$percent_change, method=c("spearman"))
cor.test(correlation$SOD, correlation$Diverse, method=c("spearman"))
cor.test(correlation$SOD, correlation$Single, method=c("spearman"))
```

Generate three panel plot.  

```{r}
sod_corr<-plot_grid(Cor1, Cor2, Cor3, labels = c("", "", ""), ncol=3, nrow=1, rel_heights= c(1,1,1), rel_widths = c(1,1,1.2), label_size = 20, label_y=1, align="h")

#ggsave(filename="Figures/SOD_correlations.png", plot=sod_corr, dpi=300, width=20, height=6, units="in")
```

## 3. Calcification  

Calcification (growth) was measured using the buoyant weight technique. We are analyzing growth as total percent growth (grams per gram per day) as reached by the end of the stress test period.  

Load in buoyant data from Excel worksheet. Loading in final calculations, manipulations conducted in Excel worksheet.     

```{r, restults=FALSE}
bwt<-read.csv("Data/Adults_Growth.csv", header=T, na.strings="NA")
bwt$Genotype<-as.factor(bwt$Colony)

#change genotypes to reflect clonal clusters 
levels(bwt$Genotype)[levels(bwt$Genotype)=="3"] <- "ClusterA"
levels(bwt$Genotype)[levels(bwt$Genotype)=="32"] <- "ClusterA"
levels(bwt$Genotype)[levels(bwt$Genotype)=="5"] <- "ClusterA"

levels(bwt$Genotype)[levels(bwt$Genotype)=="2"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="31"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="12"] <- "ClusterC"
levels(bwt$Genotype)[levels(bwt$Genotype)=="24"] <- "ClusterC"

bwt_stress<-bwt[which(bwt$Phase == "Stress"),]
```

Calculate mean values.  

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
bwt_stress<-bwt_stress %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#calculate means
selection_bwt <- plyr::ddply(bwt_stress, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(TotalPercentDay, na.rm=TRUE))

#spread by diversity treatment
selection_bwt<-selection_bwt %>% 
  spread(Treatment, mean)

#output data for analysis
selection_bwt$Metric<-c("Growth")
selection_bwt$Stage<-c("Adult")

write.csv(selection_bwt, "Output/adult_bwt_means.csv")
```

### Plotting  

Plot bwt for each genotype in each treatment.  
```{r}
bwt_summary<-bwt_stress%>%
  group_by(Genotype, Treatment, Temperature)%>%
  filter(!TotalPercentDay> 1.1)%>%
  summarise(mean=mean(TotalPercentDay, na.rm=TRUE), sd=sd(TotalPercentDay, na.rm=TRUE), N=length(TotalPercentDay[!is.na(TotalPercentDay)]), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

bwt_plot<-ggplot(data=bwt_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") + 
    ylab(expression(bold(paste("% growth day"^-1))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); bwt_plot

ggsave("Figures/Adult_bwt.pdf", plot=bwt_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```


### Analysis  

Conduct linear mixed effect model to analyze the effect of temperature treatment, diversity, and genotype on growth. 
 
```{r}
bwt_analysis<-bwt_stress%>%
  filter(Phase=="Stress")%>%
  filter(!TotalPercentDay> 1.1) # removing outlier

model_growth<-lmer(TotalPercentDay~Temperature * Treatment * Genotype + (1|Plate) + (1|Genotype) + (1|Table), data=bwt_analysis)
qqPlot(residuals(model_growth))
anova(model_growth, type=2)

leveneTest(residuals(model_growth)~Temperature*Treatment, data=bwt_analysis)
```

Analyze with genotype only as a random effect.  

```{r}
model_growth_b<-lmer(TotalPercentDay~Temperature * Treatment + (1|Plate) + (1|Genotype) + (1|Table), data=bwt_analysis)
qqPlot(residuals(model_growth_b))
anova(model_growth_b, type=2)

leveneTest(residuals(model_growth_b)~Temperature*Treatment, data=bwt_analysis)
```

## 4. PAM    

Photosynthetic efficiency (measured by PAM method). This value is collected as maximum quantum yield (Fv/Fm) for each fragment during the stress test. Here, we use calcualtions of PAM at the end of the stress test. 

Load in PAM datasets and match the record number to sample.     

```{r, results=FALSE, warning=FALSE, message=FALSE}
#load sample id and master data
pam<-read.csv("Data/PAM.csv", sep=",", na.strings="NA")
pam$Genotype<-as.factor(pam$Colony)
pam$Date<-as.factor(pam$Date)
pam$Sample<-paste(pam$Date, pam$ID)

#load pam data
t1<-read.csv("Data/T1_61019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t1$Date <- c("20190610")
t1<-plyr::rename(t1, c("V4"="ID", "V10"="FvFm"))
t1$Sample<-paste(t1$Date, t1$ID)

t2<-read.csv("Data/T2_61719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t2$Date <- c("20190617")
t2<-plyr::rename(t2, c("V4"="ID", "V10"="FvFm"))
t2$Sample<-paste(t2$Date, t2$ID)

t3<-read.csv("Data/T3_62019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t3$Date <- c("20190620")
t3<-plyr::rename(t3, c("V4"="ID", "V10"="FvFm"))
t3$Sample<-paste(t3$Date, t3$ID)

t4<-read.csv("Data/T4_62419.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t4$Date <- c("20190624")
t4<-plyr::rename(t4, c("V4"="ID", "V10"="FvFm"))
t4$Sample<-paste(t4$Date, t4$ID)

t5<-read.csv("Data/T5_62719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t5$Date <- c("20190627")
t5<-plyr::rename(t5, c("V4"="ID", "V10"="FvFm"))
t5$Sample<-paste(t5$Date, t5$ID)

t6<-read.csv("Data/T6_70119.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t6$Date <- c("20190701")
t6<-plyr::rename(t6, c("V4"="ID", "V10"="FvFm"))
t6$Sample<-paste(t6$Date, t6$ID)

#combine all pam data files 
data<-rbind(t1,t2,t3,t4,t5, t6)

#match FvFm values by date and sample number
pam$FvFm<-data$FvFm[match(pam$Sample, data$Sample)]
pam$FvFm<-as.numeric(pam$FvFm)

#match colony to genotype cluster
levels(pam$Genotype)[levels(pam$Genotype)=="3"] <- "ClusterA"
levels(pam$Genotype)[levels(pam$Genotype)=="32"] <- "ClusterA"
levels(pam$Genotype)[levels(pam$Genotype)=="5"] <- "ClusterA"

levels(pam$Genotype)[levels(pam$Genotype)=="2"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="31"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="12"] <- "ClusterC"
levels(pam$Genotype)[levels(pam$Genotype)=="24"] <- "ClusterC"
```

Calculate a percent change for each fragment over the course of the stress test.      

```{r, results=FALSE, warning=FALSE}
initial<-subset(pam, Date=="20190610", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm, Genotype))
final<-subset(pam, Date=="20190701", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm, Genotype))

totalPAM1<-rbind(initial, final)

totalPAM1<-spread(totalPAM1, StressDay, FvFm)

totalPAM1$Change<-(totalPAM1$`20`-totalPAM1$`1`)/totalPAM1$`1`

#remove colony 23, which only had one frag observation
totalPAM<-totalPAM1[!totalPAM1$Frag == "1_23_14", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_49", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_57", ]

```

Generate mean values for calculations.     

```{r, results=TRUE}
#add column for unique counts to insert the number of genotypes
totalPAM<-totalPAM %>%
  mutate(N_Genotypes = length(unique(Genotype)))

#calculate means
selection_pam <- plyr::ddply(totalPAM, c("Genotype", "Treatment", "Temperature", "N_Genotypes"), summarise,
                           mean = mean(Change, na.rm=TRUE))

#spread by diversity treatment
selection_pam<-selection_pam %>% 
  spread(Treatment, mean)

#output data for analysis
selection_pam$Metric<-c("Photosynthetic Efficiency")
selection_pam$Stage<-c("Adult")

write.csv(selection_pam, "Output/adult_pam_means.csv")
```

### Plotting  

Plot survivorship for each genotype in each treatment.  
```{r}
pam_summary<-totalPAM%>%
  group_by(Genotype, Treatment, Temperature)%>%
  summarise(mean=mean(Change, na.rm=TRUE), sd=sd(Change, na.rm=TRUE), N=length(Change[!is.na(Change)]), se=sd/sqrt(N))%>%
  mutate(code=paste(Treatment, Temperature))

pam_plot<-ggplot(data=pam_summary, aes(x = Temperature, y = mean, color = Treatment, shape=Treatment, linetype=Treatment, group=interaction(Treatment, Temperature))) +
    geom_point(size=3, position=position_dodge(0.3), color="black") +
   geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=0, linetype="solid", position=position_dodge(0.3), size=1.3, color="black") +
  facet_grid(~Genotype)+
  geom_line(position=position_dodge(0.3), size=1.3, aes(group=Treatment), color="black") +
  scale_shape_manual(values=c(1, 19))+
  scale_linetype_manual(values=c("dotted", "solid"))+
    #scale_color_manual(values = c("red4", "red2", "blue", "lightblue"), limits=c("Diverse High", "Single High", "Diverse Ambient", "Single Ambient"), labels=c("Diverse-High", "Single-High", "Diverse-Ambient", "Single-Ambient"))+
    xlab("Temperature Treatment") +
    ylim(-0.3, 0.2)+
    ylab(expression(bold(paste("% change in Fv/Fm"))))+
    theme_classic() + 
    theme(
      legend.title=element_text(face="bold", size=14),
      legend.text=element_text(size=14),
      axis.title=element_text(face="bold", size=14),
      axis.text=element_text(size=12, color="black"), 
      strip.text.x=element_text(face="italic", size=14)
      ); pam_plot

ggsave("Figures/Adult_PAM.pdf", plot=pam_plot, height=4, width=12, units = c("in"), dpi=300) #output figure
```

### Analysis  

Conduct linear mixed effect model to analyze the effect of temperature treatment, diversity, and genotype on growth.  
 
```{r}
pam_analysis<-totalPAM

model_pam<-lmer(Change~Temperature * Treatment * Genotype + (1|Plate) + (1|Genotype) + (1|Table), data=pam_analysis)
qqPlot(residuals(model_pam))
anova(model_pam, type=2)

#leveneTest(residuals(model_pam)~Temperature*Treatment*Genotype, data=pam_analysis)
``` 

Analyze with genotype as a random effect only.     
```{r}
model_pam_b<-lmer(Change~Temperature * Treatment + (1|Plate) + (1|Genotype) + (1|Table), data=pam_analysis)
qqPlot(residuals(model_pam_b))
anova(model_pam_b, type=2)

#leveneTest(residuals(model_pam_b)~Temperature*Treatment, data=pam_analysis)
```

## Combine adult figures  

```{r}
adultfig<-plot_grid(surv_plot, bwt_plot, pam_plot, labels=c("A", "B", "C"), label_size=22,  label_fontface = "bold", ncol=1, nrow=3, rel_heights= c(1,1,1), rel_widths = c(1,1,1), align="vh")

ggsave(filename="Figures/Adult_Figure.png", plot=adultfig, dpi=500, width=14, height=12, units="in")
```

# **Combine all data sets**  
 
```{r}
selection_values<-dplyr::bind_rows(selection_surv, selection_settle)
selection_values<-dplyr::bind_rows(selection_values, selection_stress)
selection_values<-dplyr::bind_rows(selection_values, selection_bwt)
selection_values<-dplyr::bind_rows(selection_values, selection_pam)

#add "experiment" and "dataset" columns to master dataframe for calculations
metadata<-read.csv("Data/metadata.csv", na.strings="NA")
selection_values_all<-left_join(selection_values, metadata)

#rename diverse column (Yoj) and single column (M) for calculations
selection_values_all<-selection_values_all%>%
  rename(Yoj=Diverse)%>%
  rename(M=Single)

write.csv(selection_values_all, "Output/selection_values_all.csv")
``` 




# **Calculate coefficients and produce figure**  
 
Calculate coefficients for selection, complementarity, and net biodiversity effects. Scripts written by C. Drury.  
 
```{r}
raw<-read.csv("Output/selection_values_all.csv")%>%mutate(Yoj=as.numeric(Yoj),M=as.numeric(M),experiment=as.factor(experiment))
#where M=uncorrected survivorship in monoculture (proportion) and Yoj=uncorreted survivorship in polyculture
out<-raw%>%group_by(experiment)%>%filter(M!="NA")%>%add_tally(name="N_Genotypes")%>% #filter missing values and recalculate number of genotypes
     mutate(Mi=M*N_Genotypes)%>%
     mutate(Yo=sum(Yoj))%>%
     mutate(Ryej=1/n())%>%
     mutate(Ryoj=Yoj/Mi)%>%
     mutate(Ryoj=case_when(Ryoj==Inf~0,TRUE~as.numeric(Ryoj)))%>% #divide by zero problem in one AHS
     mutate(Yej=Mi*Ryej)%>%
     mutate(Ye=sum(Yej))%>%
     mutate(deltaY=Yo-Ye)%>%
     mutate(deltaRY=Ryoj-Ryej)%>%
     mutate(C=n()*mean(Mi)*mean(deltaRY))%>%
     mutate(S=cov(deltaRY,Mi)*n())%>%
     mutate(N=S+C)%>%
     slice(1)%>%select(experiment,S,C,N)%>%
     left_join(.,raw%>%select(experiment,Temperature,Stage,Metric)%>%distinct(),by="experiment")
```
 
Build figure of coefficients.  

```{r}
var_width = 20
plot<-out%>%gather(compartment,coefficient,-Temperature,-experiment,-Metric,-Stage)%>%
     group_by(Temperature,Metric,Stage,compartment)%>%mutate(mean=mean(coefficient),sd=sd(coefficient))%>%
     mutate(Metric=str_wrap(Metric, width = var_width))

plot$compartment<-factor(plot$compartment,levels=c("S","C","N"),labels=c("S","C","Net"))

tag_facet2 <- function(p, open = "(", close = ")", tag_pool = letters, x = -Inf, y = Inf, 
                       hjust = -0.5, vjust = 1.5, fontface = 2, family = "", ...) {
     
     gb <- ggplot_build(p)
     lay <- gb$layout$layout
     tags <- cbind(lay, label = paste0(open, tag_pool[lay$PANEL], close), x = x, y = y)
     p + geom_text(data = tags, aes_string(x = "x", y = "y", label = "label"), ..., hjust = hjust, 
                   vjust = vjust, fontface = fontface, family = family, inherit.aes = FALSE)
}

adult<-ggplot(plot%>%filter(Stage=="Adult"))+geom_hline(yintercept=0,color="lightgray",linetype="dashed")+
     geom_point(aes(compartment,mean,color=Temperature),size=2)+geom_errorbar(aes(compartment,ymin=mean-sd,ymax=mean+sd),width=0)+
     facet_wrap(~Metric,scale="free_x")+
     theme_classic(base_size=8)+
     scale_color_manual(values=c("Blue","Red"))+
     theme(axis.title.x=element_blank(),
           legend.key.size=unit(0.2,"cm"),
           axis.text.y=element_blank(),
           axis.line.y=element_blank(),
           axis.ticks.y=element_blank(),
           axis.title.y=element_blank())+
     scale_y_continuous(limits=c(-0.7,1.1),breaks=seq(-0.7,1.1,0.2))
my_tag<-c("Adult","","")
adult<-tag_facet2(adult, 
              x = 1, y = 1, 
              vjust = 0, hjust = 0,
              open = "", close = "",
              fontface = 'italic',
              size = 3,
              tag_pool = my_tag)

larvae<-ggplot(plot%>%filter(Stage=="Larvae"))+geom_hline(yintercept=0,color="lightgray",linetype="dashed")+
     geom_point(aes(compartment,mean,color=Temperature),position = position_dodge(width = 0.5))+
     geom_errorbar(aes(compartment,ymin=mean-sd,ymax=mean+sd,color=Temperature),position=position_dodge(width=.5),width=0.1)+
     facet_wrap(~Metric)+
     theme_classic(base_size=8)+
     scale_color_manual(values=c("Blue","Red"))+
     ylab("Coefficient")+
     theme(axis.title.x=element_blank(),
           legend.position="none")+
     scale_y_continuous(limits=c(-0.7,1.1),breaks=seq(-0.7,1.1,0.2))
my_tag<-c("Larvae","")
larvae<-tag_facet2(larvae, 
                  x = 1, y = 1, 
                  vjust = 0, hjust = 0,
                  open = "", close = "",
                  fontface = 'italic',
                  size = 3,
                  tag_pool = my_tag)

quartz(h=2,w=5.2)
plot_grid(larvae,adult,rel_widths=c(2,3.2),labels=c("A","B"),label_size=8,label_x=c(0.1,-0.02))

diversity_fig<-plot_grid(larvae,adult,rel_widths=c(2,3.2),labels=c("A","B"),label_size=8,label_x=c(0.1,-0.02), ncol=2, nrow=1, rel_heights= c(1.2,1),  label_y=1, align="h")

ggsave(filename="Figures/Diversity.pdf", plot=diversity_fig, dpi=300, width=5.2, height=2, units="in")
ggsave(filename="Figures/Diversity.png", plot=diversity_fig, dpi=300, width=5.2, height=2, units="in")
```

Build figure for SOD datasets.  

```{r}
depth<-read_tsv("Data/SOD.counts")%>%select(-X11)%>%mutate(mean=rowMeans(.[,1:10]))
loci<-read_tsv("Data/SOD.beagle")%>%select(marker,allele1,allele2)
cols<-colnames(read_tsv("Data/SOD.beagle")%>%select(marker,contains('Ind')))

rawdat<-read_tsv("Data/SOD.geno",col_names=FALSE)%>%select(-X33)%>%unite(marker,X1,X2,sep="_")
colnames(rawdat)<-cols

format<-rawdat%>%select(marker,contains('_'))%>%
     rowwise()%>%
     mutate(AH12=Ind0_1+(2*Ind0_2))%>%select(-Ind0_1,-Ind0_2)%>%
     mutate(AH1=Ind1_1+(2*Ind1_2))%>%select(-Ind1_1,-Ind1_2)%>%
     mutate(AH23=Ind2_1+(2*Ind2_2))%>%select(-Ind2_1,-Ind2_2)%>%
     mutate(AH24=Ind3_1+(2*Ind3_2))%>%select(-Ind3_1,-Ind3_2)%>%
     mutate(AH2=Ind4_1+(2*Ind4_2))%>%select(-Ind4_1,-Ind4_2)%>%
     mutate(AH31=Ind5_1+(2*Ind5_2))%>%select(-Ind5_1,-Ind5_2)%>%
     mutate(AH32=Ind6_1+(2*Ind6_2))%>%select(-Ind6_1,-Ind6_2)%>%
     mutate(AH3=Ind7_1+(2*Ind7_2))%>%select(-Ind7_1,-Ind7_2)%>%
     mutate(AH4=Ind8_1+(2*Ind8_2))%>%select(-Ind8_1,-Ind8_2)%>%
     mutate(AH5=Ind9_1+(2*Ind9_2))%>%select(-Ind9_1,-Ind9_2)%>%
     gather(sample,GL,-marker)%>%
     separate(marker,into=c("CHR","POS"),sep=12)%>%
     mutate(POS=as.numeric(POS))%>%
     filter(POS>46503&POS<46620)

surv<-read_csv("Output/Colony_Surival_Change.csv")%>%filter(Temperature=="High")%>%select(Colony,percent_change)%>%
     mutate(AH="AH")%>%
     unite(sample,AH,Colony,sep="")

surv_disadvantage<-dplyr::left_join(format,surv,by="sample")%>%mutate(genotype=case_when(GL<0.31~"AA",TRUE~"AB"))%>%
     mutate(percent_change=percent_change*-1)

activity<-read_xlsx("Data/SOD_activity.xlsx",sheet="R")%>%drop_na()%>%
     separate(sample,into=c("time","colony","plate"),sep="_")%>%
     group_by(colony)%>%mutate(mean=mean(SOD))%>%
     mutate(ex="AH")%>%unite(sample,ex,colony,sep="",remove=FALSE)%>%
     left_join(.,surv_disadvantage,by="sample")%>%select(-ex)%>%
     mutate(SOD=SOD/30)
wilcox.test(SOD~genotype,data=activity) #analyze difference in SOD

a<-ggplot(surv_disadvantage)+
     #geom_boxplot(aes(genotype,percent_change,fill=genotype))+
     geom_point(aes(genotype,percent_change,fill=genotype),pch=21)+
     ylab("Isolated Survivorship Disadvantage")+xlab("Genotype\n(SczhEng_501:46602)")+
     theme_classic(base_size=8)+
     theme(legend.position="none")+
     scale_fill_manual(values=c("orange","gray"))

b<-ggplot(activity)+geom_boxplot(aes(reorder(colony,mean),SOD,fill=genotype))+
     geom_point(aes(reorder(colony,mean),SOD,fill=genotype),pch=21)+
     theme_classic(base_size=8)+
     xlab("Colony")+ylab("SOD Activity (U/μg)")+
     scale_fill_manual(values=c("orange","gray"))+
     scale_color_manual(values=c("orange","gray"))+
     theme(legend.position="none",
           axis.text.x=element_text(angle=90,hjust=1,vjust=0.5))+
     annotate("text",x=1,y=0.08,label="Wilcox ~Genotype p=0.092",size=2,fontface="italic",color="black",hjust=0)

quartz(w=3.74,h=2.294)
sod_plot<-plot_grid(a,b,align="h",axis="tb",rel_widths=c(2,3),nrow=1,labels=c("A","B"),label_size=8,label_x=c(0.1,0)) 

ggsave(filename="Figures/SOD.pdf", plot=sod_plot, dpi=300, width=3.74, height=2.294, units="in")
ggsave(filename="Figures/SOD.png", plot=sod_plot, dpi=300, width=3.74, height=2.294, units="in")
```



# **Temperature and light during adult experiment**  

Plot temperature data during stress phase of experiment.   
```{r}
library(readxl);library(tidyverse);library(lubridate);library(scales);library(plotrix)
data<-read_csv("Data/Light_temp.csv");str(temps)

data$Treatment<-as.factor(data$Treatment)
data$Table<-as.factor(data$Table)
#temps$Date<-as.POSIXct(temps$DateTime, format-)
data$Date <- format(as.POSIXct(strptime(data$DateTime,"%m/%d/%y %H:%M",tz="")) ,format = "%m/%d")
data$Date <- as.POSIXct(data$DateTime, "%m/%d/%y %H:%M",tz="")

temps<-data%>%
     filter(Phase=="Stress")

temp_plot<-ggplot(temps)+geom_point(aes(Date,Temp,color=Treatment),alpha=0.3,size=1)+
     theme_classic()+
     scale_color_manual(values=c("Blue","Red"))+
     geom_smooth(aes(Date,Temp,color=Treatment),se=FALSE,span=0.15,size=0.5)+
     scale_x_datetime(date_breaks="7 days",minor_breaks=waiver(),labels=date_format("%m-%d"))+
     #scale_y_continuous(breaks=seq(23,32,1))+
     ylim(22, 33)+
     ylab("Temperature (°C)")+
     theme(legend.key.size = unit(1,"line"))+
    theme(legend.position="none")+
    theme(legend.title=element_text(size=10))+
    theme(plot.margin = margin(0.5, 0.1, 0.1, 0.1, "cm"))+
    theme(legend.text=element_text(size=9)); temp_plot

#ggsave("Figures/temperature_plot.png", plot=temp_plot, w=5, h=3, units="in", dpi=300)

light_plot<-ggplot(temps)+geom_point(aes(Date,LightLux,color=Treatment),alpha=0.3,size=1)+
     theme_classic()+
     scale_color_manual(values=c("Blue","Red"))+
     geom_smooth(aes(Date,LightLux,color=Treatment),se=FALSE,span=0.15,size=0.5)+
     scale_x_datetime(date_breaks="7 days",minor_breaks=waiver(),labels=date_format("%m-%d"))+
     #scale_y_continuous(breaks=seq(23,32,1))+
     ylim(0, 150000)+
     ylab("Light (Lux)")+
     theme(legend.key.size = unit(1,"line"))+
    theme(legend.position="none")+
    theme(legend.title=element_text(size=10))+
    theme(plot.margin = margin(0.5, 0.1, 0.1, 0.1, "cm"))+
    theme(legend.text=element_text(size=9)); light_plot


#create a legend for later 
#build a plot for the legend
legend_plot<-temp_plot + 
  theme(legend.position="right")

# extract the legend from one of the plots
legend <- get_legend(
  # create some space to the left of the legend
  legend_plot)

#+ theme(legend.box.margin = margin(1,1,1,1))
#)

```

Display grow-out period temperatures and light conditions.  
```{r}
growout<-data%>%
  filter(Phase=="Growth")

growtemp_plot<-ggplot(growout)+geom_point(aes(Date,Temp,color=Treatment),alpha=0.3,size=1)+
     theme_classic()+
     scale_color_manual(values=c("Blue","Red"))+
     geom_smooth(aes(Date,Temp,color=Treatment),se=FALSE,span=0.15,size=0.5)+
     scale_x_datetime(date_breaks="14 days",minor_breaks=waiver(),labels=date_format("%m-%d"))+
     #scale_y_continuous(breaks=seq(23,32,1))+
     ylim(22, 33)+
     ylab("Temperature (°C)")+
     theme(legend.key.size = unit(1,"line"))+
    theme(legend.position="none")+
    theme(legend.title=element_text(size=10))+
    theme(plot.margin = margin(0.5, 0.1, 0.1, 0.1, "cm"))+
    theme(legend.text=element_text(size=9)); growtemp_plot

growlight_plot<-ggplot(growout)+geom_point(aes(Date,LightLux,color=Treatment),alpha=0.3,size=1)+
     theme_classic()+
     scale_color_manual(values=c("Blue","Red"))+
     geom_smooth(aes(Date,LightLux,color=Treatment),se=FALSE,span=0.15,size=0.5)+
     scale_x_datetime(date_breaks="14 days",minor_breaks=waiver(),labels=date_format("%m-%d"))+
     #scale_y_continuous(breaks=seq(23,32,1))+
     ylim(0, 150000)+
     ylab("Light (Lux)")+
     theme(legend.key.size = unit(1,"line"))+
    theme(legend.position="none")+
    theme(legend.title=element_text(size=10))+
    theme(plot.margin = margin(0.5, 0.1, 0.1, 0.1, "cm"))+
    theme(legend.text=element_text(size=9)); growlight_plot
```


Display mean values of stress period and growth period temperatures.   
```{r}
temps%>%group_by(Treatment)%>%summarise(mean=mean(Temp, na.rm=TRUE),sd=sd(Temp, na.rm=TRUE),se=std.error(Temp, na.rm=TRUE))

data%>%filter(Phase=="Growth")%>%group_by(Treatment)%>%summarise(mean=mean(Temp, na.rm=TRUE),sd=sd(Temp, na.rm=TRUE),se=std.error(Temp, na.rm=TRUE))

```


Test for differences in temperature and light between treatments.  

```{r}
t.test(data=temps, Temp~Treatment) #p<0.001
t.test(data=temps, LightLux~Treatment) #p=0.670
```


Assemble final environmental figure.  

```{r}
library(cowplot)

environ_panel<-plot_grid(growtemp_plot, temp_plot, growlight_plot,light_plot, labels=c("A", "B", "C", "D"), label_size=14, ncol=2, nrow=2, rel_heights= c(1,1, 1, 1), rel_widths = c(1,1, 1, 1), label_y=1, align="vh")

# add the legend to the row we made earlier. Give it one-third of 
# the width of one plot (via rel_widths).
environ_plot_legend<-plot_grid(environ_panel, legend, rel_widths = c(6, .6), ncol=2, nrow=1)
 
ggsave(filename="Figures/Full_Environ_Figure.png", plot=environ_plot_legend, dpi=500, width=12, height=8, units="in")
```