---
title: "Pocillopora acuta Intraspecific Diversity Effects: Larvae, Recruits, Adults"
author: "Ariana Huffmyer (R Markdown author), Nina Bean, Casey Harris, Crawford Drury"
date: "2020"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 6
    toc_float: yes
  pdf_document:
    keep_tex: yes
---

## Setup  

Set up workspace, set options, and load required packages.
```{r, echo=FALSE, show=FALSE}
rm(list=ls(all=TRUE)) 
```

```{r, echo=FALSE, warning=FALSE, results=FALSE, message=FALSE}
library(multcomp)
library(ggplot2)
library(dplyr)
library(plyr)
library(lattice)
library(MASS)
library(car)
library(FSA)
library(tidyverse)
library(partitionBEFsp)
library(factoextra)
library(ggfortify)
library(cowplot)
library(readxl)
library(ggdendro)
library(ape)
library(SNPRelate)
```
 
# **Genotyping**  

We used next generation sequencing to obtain 150bp paired-end reads of DNA from all parent colonies in larvae, juvenile, and adult experiments. Scripts of sequence analysis are available in the R Project as a text file. After obtaining an identity-by-distance (IBS) matris with the ANGSD package in Python, we produced a tree using the hclust function with "complete" clustering.  

Produce an MDS plot and tree with ANGSD data. Blue line indicates cut-off for clonal groups based on distance between biological replicates.  
```{r, results=TRUE}
#read IBS matrix
setwd("Data/")
name <- "angsd.IBS.ibsMat"
m <- as.matrix(read.table(name))

#replace labels with sample names
samples<-read.table("sample_list.txt");samples<-as.vector(samples$V1)
colnames(m)<-samples
rownames(m)<-samples

#mds plot
mds <- cmdscale(as.dist(m))
plot(mds,lwd=2,ylab="Dist",xlab="Dist",main="multidimensional scaling",col=rep(1:32))

plot(hclust(dist(m), "complete"))
abline(h=0.01485, col="blue") #genetic distance b/t SM29 and SM41
```

Generate final figure with colors representing clonal groups.  
```{r, results=TRUE, message=FALSE, warning=FALSE}
rawdata<-read_tsv("Data/angsd.IBS.ibsMat",col_names=FALSE)%>%select(-X37)

hcdata<-dendro_data(hclust(dist(rawdata),"complete"),type="rectangle")
hcdata$segments<-hcdata$segments%>%mutate(yend=case_when(yend==0~(y-0.003),TRUE ~ (yend)))
hcdata$labels<-left_join(hcdata$labels,hcdata$segments,by="x")%>%select(-y.x)%>%dplyr::rename(y=yend)%>%mutate(y=y-.01)
hcdata$segments<-left_join(hcdata$segments,hcdata$labels,by="x")%>%select(-y.y,-xend.y,-y.y.y)%>%dplyr::rename(y=y.x,xend=xend.x)%>%
     mutate(color=case_when(label=="8"~"a",
                            label=="31"~"a",
                            label=="29"~"a",
                            label=="7"~"a",
                            label=="10"~"a",
                            label=="17"~"a",
                            label=="32"~"a",
                            label=="11"~"b",
                            label=="13"~"b",
                            label=="18"~"b",
                            label=="5"~"c",
                            label=="6"~"c",
                            label=="1"~"c",
                            label=="4"~"c",
                            label=="28"~"d",
                            label=="12"~"d",
                            label=="16"~"d",
                            TRUE ~ ("e")))

#quartz(width=1.75,height=1.5)
Figure1<-ggplot() + 
     geom_hline(yintercept=0.0148,color="gray",linetype="dashed")+
     geom_segment(data=segment(hcdata), aes(x=x, y=y, xend=xend, yend=yend,colour=color),size=.5) +
     #scale_y_continuous(breaks=seq(-.002,.02,.002),limits=c(-.02,.02))+
     theme_classic()+
     theme(axis.ticks.x=element_blank(),
           axis.text.x=element_blank(),
           axis.line.x=element_blank(),
           axis.title.x=element_blank(),
           legend.position="none")+
     ylab("Genetic Dist")+
     scale_color_manual(values=c("red","blue","orange","darkgreen","black"));Figure1
```

```{r, results=FALSE}
ggsave(filename="Figures/Figure1.pdf", plot=Figure1, dpi=300, width=1.75, height=1.5, units="in")

```
These clonal groups are accounted for in below analysis in which number of genotypes is at times lower than the number of colonies due to clonal groupings.  


# **Larval Experiment**  

In this experiment with larvae, we exposed larvae of P. acuta either to single (3 larvae from same parent colony) or multiple (1 larvae from each of 3 different parent colonies) to ambient or high temperature. Larvae were put in 96-well plates and treated with temperature using a heat block. Larvae were not exposed to sunlight or artificial LED light.  

## 1. Load and manipulate data sets  

First, load survivorship data frame. Survival was recorded for each larvae at 2 and 4 days of temperature treatment exposure. Data was collected at the level of the individual larva. Here, we also reassign colony identity to clonal identity to account for number of distinct individuals.    

```{r, results=FALSE}
surv<-read.csv("Data/larval_surv.csv", sep=",", na.strings="NA")
surv$Plate<-as.factor(surv$Plate)
surv$Well<-as.factor(surv$Well)
surv$Column<-as.factor(surv$Column)
surv$Row<-as.factor(surv$Row)
surv$site<-as.factor(surv$site)
surv$colony<-as.factor(surv$colony)
surv$diversity_level<-as.factor(surv$diversity_level)
surv$diversity_level<-relevel(surv$diversity_level, ref="1")

levels(surv$colony)[levels(surv$colony)=="831"] <- "ClusterA"
levels(surv$colony)[levels(surv$colony)=="821"] <- "ClusterA"
levels(surv$colony)[levels(surv$colony)=="801"] <- "ClusterA"
levels(surv$colony)[levels(surv$colony)=="841"] <- "ClusterA"

levels(surv$colony)[levels(surv$colony)=="741"] <- "ClusterB"
levels(surv$colony)[levels(surv$colony)=="743"] <- "ClusterB"
levels(surv$colony)[levels(surv$colony)=="802"] <- "ClusterB"

levels(surv$colony)[levels(surv$colony)=="815"] <- "ClusterD"
levels(surv$colony)[levels(surv$colony)=="742"] <- "ClusterD"
levels(surv$colony)[levels(surv$colony)=="750"] <- "ClusterD"
```

Then, assign Success and Failure integer columns. Success (1) = alive and Failure (0) = dead. Subset data to analyze survivorship at the end of 4-day treatment period.  

```{r, results=FALSE}
#assign outcome as two columns: success and failure
surv$success<-ifelse(surv$outcome==1,1,0)
surv$failure<-ifelse(surv$outcome==0,1,0)

#provide subset for each day
midsurv <- surv[which(surv$duration == "2"),]
finalsurv <- surv[which(surv$duration == "4"),]
```

Load the settlement data frame and assign Success and Failure columns as above for survivorship. Settlement was also recorded at the level of the individual larva and tracked at 2 and 4 days of treatment exposure.   

```{r, results=FALSE}
settle<-read.csv("Data/larval_settle.csv", sep=",", na.strings="NA")
settle$Date<-as.factor(settle$Date)
settle$Plate<-as.factor(settle$Plate)
settle$Well<-as.factor(settle$Well)
settle$Column<-as.factor(settle$Column)
settle$Row<-as.factor(settle$Row)
settle$site<-as.factor(settle$site)
settle$colony<-as.factor(settle$colony)
settle$diversity_level<-as.factor(settle$diversity_level)
settle$diversity_level<-relevel(settle$diversity_level, ref="1")

settle$success<-ifelse(settle$outcome==0,1,0)
settle$failure<-ifelse(settle$outcome==1,1,0)

levels(settle$colony)[levels(settle$colony)=="831"] <- "ClusterA"
levels(settle$colony)[levels(settle$colony)=="821"] <- "ClusterA"
levels(settle$colony)[levels(settle$colony)=="801"] <- "ClusterA"
levels(settle$colony)[levels(settle$colony)=="841"] <- "ClusterA"

levels(settle$colony)[levels(settle$colony)=="741"] <- "ClusterB"
levels(settle$colony)[levels(settle$colony)=="743"] <- "ClusterB"
levels(settle$colony)[levels(settle$colony)=="802"] <- "ClusterB"

levels(settle$colony)[levels(settle$colony)=="815"] <- "ClusterD"
levels(settle$colony)[levels(settle$colony)=="742"] <- "ClusterD"
levels(settle$colony)[levels(settle$colony)=="750"] <- "ClusterD"

#provide subset for each day
midsettle <- settle[which(settle$duration == "2"),]
finalsettle <- settle[which(settle$duration == "4"),]
```
  
## 2. Selection and Complementarity  

Calculating selection and complementarity by Loreau and Hector method (2001) with the "partitionBEFsp" package.    

First, we assemble data frame that has value for each genotype/clonal group in single and diverse groups at high and ambient temperature. Data collected at 4-days of treatment exposure.  
 
Calculate population level values for 1) survivorship at ambient, 2) survivorship at high, 3) settlement at ambient, 4) settlement at high. Since we are sampling the complete population, the number of genotypes in the sample (N) and population (Q) are the same. As the calculations are based on a ratio of N and P, values are the same when the absolute values of N and P change but remain equal. We are using the "classic" calculation of diversity effects to produce values for selection (S), complementarity (C), and net effects (Y), which is the combined effect of S and C.  

Steps in calculation are as follows: 

- Generate a data frame with mean value of metric of interest for each genotype in single and diverse treatments at ambient and high temperature.  

- Use "calculate_DRY" function to calculate change in relative yield for each genotype between single and multiple genotype exposures. 

- Use "classic_partition" function to calculate S, C values sensu Loreau & Hector 2001. 

- Sum S and C values to calculate Y.   

Next, display sample sizes for survivorship and then settlement.   
```{r, results=TRUE}
#survivorship
selection_surv <- ddply(finalsurv, c("Plate", "colony", "diversity_level", "Temperature"), summarise,
                           mean = mean(outcome, na.rm=TRUE))
#generate sample size
ddply(finalsurv, c("diversity_level", "Temperature", "colony"), summarise,
                            N    = length(outcome[!is.na(outcome)]))

selection_surv<-spread(selection_surv, diversity_level, mean)
selection_surv<-selection_surv[complete.cases(selection_surv), ]

#subset by temperature
L1<-selection_surv[which(selection_surv$Temperature == "Ambient"),]
L2<-selection_surv[which(selection_surv$Temperature == "High"),]

#settlement
selection_settle <- ddply(finalsettle, c("Plate", "colony", "diversity_level", "Temperature"), summarise,
                           mean = mean(outcome, na.rm=TRUE))

#generate sample size
ddply(finalsettle, c("diversity_level", "Temperature", "colony"), summarise,
                            N    = length(outcome[!is.na(outcome)]))

selection_settle<-spread(selection_settle, diversity_level, mean)
selection_settle<-selection_settle[complete.cases(selection_settle), ]

#subset by temperature
sL1<-selection_settle[which(selection_settle$Temperature == "Ambient"),]
sL2<-selection_settle[which(selection_settle$Temperature == "High"),]
```

Calculate values for larval survivorship.  
```{r, results=FALSE}

#calculate coefficients
dryLs1<-calculate_DRY(L1$`3`, L1$`1`, Q=3)
cpLs1<-classic_partition(dryLs1, L1$`1`, N=3, Q=3)
cpLs1<-as.data.frame(cpLs1)
cpLs1$Temperature<-c("Ambient")

#calculate coefficients
dryLs2<-calculate_DRY(L2$`3`, L2$`1`, Q=5)
cpLs2<-classic_partition(dryLs2, L2$`1`, N=5, Q=5)
cpLs2<-as.data.frame(cpLs2)
cpLs2$Temperature<-c("High")

surv_coefs_short<-rbind(cpLs1, cpLs2)

#calculate total biodiveristy effect
surv_coefs_short$Y<-surv_coefs_short$S + surv_coefs_short$C
surv_coefs<-gather(surv_coefs_short, Coefficient, Value, c(S,C,Y), factor_key=TRUE)
surv_coefs$Lifestage<-c("Larvae")
surv_coefs$Metric<-c("Survivorship")
``` 

Calculate values for larval settlement.  
```{r, results=FALSE}

#calculate coefficients
dryL1<-calculate_DRY(sL1$`3`, sL1$`1`, Q=3)
cpL1<-classic_partition(dryL1, sL1$`1`, N=3, Q=3)
cpL1<-as.data.frame(cpL1)
cpL1$Temperature<-c("Ambient")

#calculate coefficients
dryL2<-calculate_DRY(sL2$`3`, sL2$`1`, Q=3)
cpL2<-classic_partition(dryL2, sL2$`1`, N=3, Q=3)
cpL2<-as.data.frame(cpL2)
cpL2$Temperature<-c("High")

settle_coefs_short<-rbind(cpL1, cpL2)

#calculate total biodiveristy effect
settle_coefs_short$Y<-settle_coefs_short$S + settle_coefs_short$C
settle_coefs<-gather(settle_coefs_short, Coefficient, Value, c(S,C,Y), factor_key=TRUE)
settle_coefs$Lifestage<-c("Larvae")
settle_coefs$Metric<-c("Settlement")
```

Generate data frame of all larval coefficients.  

```{r, results=FALSE}
larval_coefs<-bind_rows(surv_coefs, settle_coefs)
```

Display coefficients.  

```{r, results=TRUE, warning=FALSE}
knitr::kable(larval_coefs)
```


# **Adult Experiment**  

Adults in this experiment were cut as fragments and exposed to either single or multiple colony groups. Single groups (n=40, n=400 fragments) included ten fragments of the same colony while multiple genotype groups (n=20 groups, n=200 fragments) included one fragment from each of ten parent colonies. In these calculations, genotype number was adjusted based on results from genetic analysis.   

Adults were exposed to ambient and high temperature treatments in outdoor flow through tank systems under one layer of shade cloth (light peaking at ~250-300 PAR daily). Corals were contained in plastic containers for each group so water was not exchanged. See manuscript for more details of methodology.  

Diversity metrics for survivorship, growth, and photosynthetic efficiency are calculated as described above.  

## 1. Survivorship    
 
Load survivorship data from the stress test period. Survival over the course of the 25 day stress test was recorded at the level of the individual fragment. In calculations we are analyzing survivorship by the end of this period.      

```{r, results=FALSE}
#load data
stress<-read.csv("Data/Adults_Survivorship.csv", header=TRUE, sep=",", na.strings="NA")
stress$Colony<-as.factor(stress$Colony)

#change genotypes to reflect clonal clusters 
levels(stress$Colony)[levels(stress$Colony)=="3"] <- "ClusterA"
levels(stress$Colony)[levels(stress$Colony)=="32"] <- "ClusterA"
levels(stress$Colony)[levels(stress$Colony)=="5"] <- "ClusterA"

levels(stress$Colony)[levels(stress$Colony)=="2"] <- "ClusterC"
levels(stress$Colony)[levels(stress$Colony)=="31"] <- "ClusterC"
levels(stress$Colony)[levels(stress$Colony)=="12"] <- "ClusterC"
levels(stress$Colony)[levels(stress$Colony)=="24"] <- "ClusterC"

#assign outcome as two columns: success and failure at the end point

endstress <- stress[which(stress$Day == "107"),]
endstress$success<-ifelse(endstress$Censor==0,1,0)
endstress$failure<-ifelse(endstress$Censor==1,1,0)
adultstress<-cbind(endstress$success, endstress$failure)

```


Selection and Complementarity calculations:  

Display sample sizes for survivorship.  
```{r, results=TRUE}

selection_stress <- ddply(endstress, c("Colony", "Treatment", "Temperature"), summarise,
                           mean = mean(success, na.rm=TRUE))

#generate sample size
ddply(endstress, c("Treatment", "Temperature", "Colony"), summarise,
                            N    = length(success[!is.na(success)]))

selection_stress<-spread(selection_stress, Treatment, mean)
selection_stress<-selection_stress[complete.cases(selection_stress), ]

selection_stressA<-selection_stress[which(selection_stress$Temperature == "Ambient"),]
selection_stressH<-selection_stress[which(selection_stress$Temperature == "High"),]
#replace a zero with 0.0001
selection_stressH$Single<-ifelse(selection_stressH$Single==0,0.1,selection_stressH$Single)

```

Calculate values for survivorship.  
```{r, results=FALSE}
#calculate coefficients for grow out, stress (ambient) and stress (high)

drystress1<-calculate_DRY(selection_stressA$Diverse, selection_stressA$Single, Q=5)
selection_stressA<-classic_partition(drystress1, selection_stressA$Single, N=5, Q=5)
selection_stressA<-as.data.frame(selection_stressA)
selection_stressA$Temperature<-c("Ambient")

drystress2<-calculate_DRY(selection_stressH$Diverse, selection_stressH$Single, Q=5)
selection_stressH<-classic_partition(drystress2, selection_stressH$Single, N=5, Q=5)
selection_stressH<-as.data.frame(selection_stressH)
selection_stressH$Temperature<-c("High")

#bind together
adult_surv_coefs_short<-rbind(selection_stressA, selection_stressH)

#calculate total biodiveristy effect
adult_surv_coefs_short$Y<-adult_surv_coefs_short$S + adult_surv_coefs_short$C
adult_surv_coefs<-gather(adult_surv_coefs_short, Coefficient, Value, c(S,C,Y), factor_key=TRUE)
adult_surv_coefs$Lifestage<-c("Adult")
adult_surv_coefs$Metric<-c("Survivorship")
```


## 2. Calcification  

Calcification (growth) was measured using the buoyant weight technique. We are analyzing growth as total percent growth (grams per gram per day) as reached by the end of the stress test period. 

Load in buoyant data from Excel worksheet. Loading in final calculations, manipulations conducted in Excel worksheet.     

```{r, restults=FALSE}
bwt<-read.csv("Data/Adults_Growth.csv", header=T, na.strings="NA")
bwt$Colony<-as.factor(bwt$Colony)
bwt$Treatment<-relevel(bwt$Treatment, ref="Single")

bwt_stress<-bwt[which(bwt$Phase == "Stress"),]
```


Selection and Complementarity calculations:  

Display sample size for growth.  

```{r, results=TRUE}
selection_stress <- ddply(bwt_stress, c("Colony", "Treatment", "Temperature"), summarise,
                           mean = mean(TotalPercentDay, na.rm=TRUE))

#generate sample size
ddply(bwt_stress, c("Treatment", "Temperature", "Colony"), summarise,
                            N    = length(TotalPercentDay[!is.na(TotalPercentDay)]))

selection_stress<-spread(selection_stress, Treatment, mean)
selection_stress<-selection_stress[complete.cases(selection_stress), ]

selection_stressA<-selection_stress[which(selection_stress$Temperature == "Ambient"),]
selection_stressH<-selection_stress[which(selection_stress$Temperature == "High"),]
```

Calculate values for survivorship.  
```{r, results=FALSE}
#calculate coefficients for stress (ambient) and stress (high)

drystress1<-calculate_DRY(selection_stressA$Diverse, selection_stressA$Single, Q=5)
selection_stressA<-classic_partition(drystress1, selection_stressA$Single, N=5, Q=5)
selection_stressA<-as.data.frame(selection_stressA)
selection_stressA$Temperature<-c("Ambient")

drystress2<-calculate_DRY(selection_stressH$Diverse, selection_stressH$Single, Q=5)
selection_stressH<-classic_partition(drystress2, selection_stressH$Single, N=5, Q=5)
selection_stressH$Temperature<-c("High")

#bind together
bwt_coefs_short<-rbind(selection_stressA, selection_stressH)

#calculate total biodiveristy effect
bwt_coefs_short$Y<-bwt_coefs_short$S + bwt_coefs_short$C
bwt_coefs<-gather(bwt_coefs_short, Coefficient, Value, c(S,C,Y), factor_key=TRUE)
bwt_coefs$Lifestage<-c("Adult")
bwt_coefs$Metric<-c("Growth")
```

## 3. PAM    

Photosynthetic efficiency (measured by PAM method). This value is collected as maximum quantum yield (Fv/Fm) for each fragment during the stress test. Here, we use calcualtions of PAM at the end of the stress test. 

Load in PAM datasets and match the record number to sample.    

```{r, results=FALSE, warning=FALSE, message=FALSE}
#load sample id and master data
pam<-read.csv("Data/PAM.csv", sep=",", na.strings="NA")
pam$Colony<-as.factor(pam$Colony)
pam$Date<-as.factor(pam$Date)
pam$Sample<-paste(pam$Date, pam$ID)

#load pam data
t1<-read.csv("Data/T1_61019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t1$Date <- c("20190610")
t1<-rename(t1, c("V4"="ID", "V10"="FvFm"))
t1$Sample<-paste(t1$Date, t1$ID)

t2<-read.csv("Data/T2_61719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t2$Date <- c("20190617")
t2<-rename(t2, c("V4"="ID", "V10"="FvFm"))
t2$Sample<-paste(t2$Date, t2$ID)

t3<-read.csv("Data/T3_62019.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t3$Date <- c("20190620")
t3<-rename(t3, c("V4"="ID", "V10"="FvFm"))
t3$Sample<-paste(t3$Date, t3$ID)

t4<-read.csv("Data/T4_62419.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t4$Date <- c("20190624")
t4<-rename(t4, c("V4"="ID", "V10"="FvFm"))
t4$Sample<-paste(t4$Date, t4$ID)

t5<-read.csv("Data/T5_62719.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t5$Date <- c("20190627")
t5<-rename(t5, c("V4"="ID", "V10"="FvFm"))
t5$Sample<-paste(t5$Date, t5$ID)

t6<-read.csv("Data/T6_70119.pam", sep=";", header=FALSE) #V4 and #V10 are sample and FV/FM values, respectively
t6$Date <- c("20190701")
t6<-rename(t6, c("V4"="ID", "V10"="FvFm"))
t6$Sample<-paste(t6$Date, t6$ID)

#combine all pam data files 
data<-rbind(t1,t2,t3,t4,t5, t6)

#match FvFm values by date and sample number
pam$FvFm<-data$FvFm[match(pam$Sample, data$Sample)]
pam$FvFm<-as.numeric(pam$FvFm)
```

Calculate a percent change for each fragment over the course of the stress test.      

```{r, results=FALSE, warning=FALSE}
initial<-subset(pam, Date=="20190610", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm))
final<-subset(pam, Date=="20190701", c(StressDay, Treatment, Colony, Plate, Frag, Table, Temperature, FvFm))

totalPAM1<-rbind(initial, final)

totalPAM1<-spread(totalPAM1, StressDay, FvFm)

totalPAM1$Change<-(totalPAM1$`20`-totalPAM1$`1`)/totalPAM1$`1`

#remove colony 23, which only had one frag observation
totalPAM<-totalPAM1[!totalPAM1$Frag == "1_23_14", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_49", ]
totalPAM<-totalPAM[!totalPAM$Frag == "2_23_57", ]

#change levels of factor
totalPAM$Treatment<-relevel(totalPAM$Treatment, ref = "Single")
```

Selection and Complementarity calculations of values at the end of the stress test period. Display sample sizes.     

```{r, results=TRUE}
selection_pam <- ddply(final, c("Colony", "Treatment", "Temperature"), summarise,
                           mean = mean(FvFm, na.rm=TRUE))

#generate sample size
ddply(final, c("Treatment", "Temperature", "Colony"), summarise,
                            N    = length(FvFm[!is.na(FvFm)]))

selection_pam<-spread(selection_pam, Treatment, mean)
selection_pam<-selection_pam[complete.cases(selection_pam), ]

selection_pamA<-selection_pam[which(selection_pam$Temperature == "Ambient"),]
selection_pamH<-selection_pam[which(selection_pam$Temperature == "High"),]

```

Calculate values for PAM.  
```{r, results=FALSE}
#calculate coefficients
dryPAM1<-calculate_DRY(selection_pamA$Diverse, selection_pamA$Single, Q=5)
pam1<-classic_partition(dryPAM1, selection_pamA$Single, N=5, Q=5)
pam1<-as.data.frame(pam1)
pam1$Temperature<-c("Ambient")

dryPAM2<-calculate_DRY(selection_pamH$Diverse, selection_pamH$Single, Q=5)
pam2<-classic_partition(dryPAM2, selection_pamH$Single, N=5, Q=5)
pam2<-as.data.frame(pam2)
pam2$Temperature<-c("High")

#combine values
pam_coefs_short<-rbind(pam1, pam2)

#add values together to get total biodiversity effect (delta Y)
pam_coefs_short$Y<-pam_coefs_short$S + pam_coefs_short$C
pam_coefs<-gather(pam_coefs_short, Coefficient, Value, c(S,C,Y), factor_key=TRUE)
pam_coefs$Lifestage<-c("Adult")
pam_coefs$Metric<-c("Photosynthetic Efficiency")
```

Combine adult data sets.  
View adult coefficients.  
```{r, results=TRUE}
adult_coefs<-bind_rows(adult_surv_coefs, bwt_coefs, pam_coefs)
```
```{r, results=TRUE}
knitr::kable(adult_coefs)
```
 
# **Export dataset**  

Combine larval, adult, and juvenile datasets and export to the "output" folder.   

```{r, results=FALSE}
diversity_coefs<-bind_rows(larval_coefs,adult_coefs) 
write.csv(diversity_coefs, file="Output/Diversity_Coefficients.csv")
```
   
```{r, results=TRUE}
knitr::kable(diversity_coefs)
```


# **Stats tests**  
# **Final Figure**  

Display final figure of all diversity coefficients. Color indicates metric, separated by life stage.  

```{r, results=TRUE, message=FALSE, warning=FALSE}
data<-read_csv("Output/Diversity_Coefficients.csv")%>%
     mutate(Coefficient=case_when(Coefficient=="S"~"Selection",
                                  Coefficient=="Y"~"Net",
                                  Coefficient=="C"~"Complementarity"))%>%
     mutate(Coefficient = factor(Coefficient, levels=c("Selection","Complementarity","Net")))%>%
     mutate(Lifestage = factor(Lifestage, levels=c("Larvae","Adult")))

#quartz(width=4.5,height=3)
Figure2<-ggplot(data)+geom_jitter(aes(Coefficient,Value,color=Metric,shape=Temperature),size=2,width=0.1,height=0,alpha=0.75)+facet_wrap(~Lifestage)+
     theme_classic(base_size=10)+
     scale_color_manual(values=c("gray","orange","#008B8B","black"))+
     scale_shape_manual(values=c(16,1))+
     geom_hline(yintercept=0,linetype="dashed",color="lightgray")+
     ylab("Coefficient Estimate")+
     xlab("")+
     scale_x_discrete(labels = c("S","C","Net"));Figure2

ggsave(filename="Figures/Figure2.pdf", plot=Figure2, dpi=300, width=4.5, height=3, units="in")
```

Calculate the proportion that complementarity values contribute to net effect.  

```{r, results=TRUE, message=FALSE, warning=FALSE}
data%>%select(-X1)%>%spread(Coefficient,Value)%>%mutate(proportion=Net/Complementarity)%>%summarise(mean=mean(proportion))
```


